
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="多组学联合分析">
      
      
      
        <meta name="author" content="xanadu314">
      
      
        <link rel="canonical" href="https://xanadu314.github.io/pathongs/blog/test0317/">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-6.2.8">
    
    
      
        <title>真核生物基因组的基因预测和注释 - HYX's Bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.cb6bc1d0.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.39b8e14a.min.css">
        
          
          
          <meta name="theme-color" content="#4051b5">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto",-apple-system,BlinkMacSystemFont,Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono",SFMono-Regular,Consolas,Menlo,monospace}</style>
      
    
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/extra.css">
    
    
      
        
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-55420348-3","auto"),ga("set","anonymizeIp",!0),ga("send","pageview"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){if(this.value){var e=document.location.pathname;ga("send","pageview",e+"?q="+this.value)}})}),document.addEventListener("DOMContentSwitch",function(){ga("send","pageview",document.location.pathname)})</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
      
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="">
      
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#_1" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid" aria-label="Header">
    <a href="https://xanadu314.github.io/pathongs/" title="HYX&#39;s Bioinformatics" class="md-header-nav__button md-logo" aria-label="HYX's Bioinformatics">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header-nav__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header-nav__title" data-md-component="header-title">
      <div class="md-header-nav__ellipsis">
        <div class="md-header-nav__topic">
          <span class="md-ellipsis">
            HYX's Bioinformatics
          </span>
        </div>
        <div class="md-header-nav__topic">
          <span class="md-ellipsis">
            
              真核生物基因组的基因预测和注释
            
          </span>
        </div>
      </div>
    </div>
    
      <label class="md-header-nav__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" data-md-component="search-reset" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header-nav__source">
        
<a href="https://github.com/indexofire/pathongs/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../.." class="md-tabs__link">
      首页
    </a>
  </li>

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../book/main/" class="md-tabs__link">
        数据分析笔记
      </a>
    </li>
  

      
        
  
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../../tutorial/Linux/01_linux-installation/" class="md-tabs__link">
        专项教程
      </a>
    </li>
  

  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../protocol/" class="md-tabs__link">
        实验手册
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../my-vim-ide/" class="md-tabs__link">
        生信笔记
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../about/" class="md-tabs__link">
        关于我们
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://xanadu314.github.io/pathongs/" title="HYX&#39;s Bioinformatics" class="md-nav__button md-logo" aria-label="HYX's Bioinformatics">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    HYX's Bioinformatics
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/indexofire/pathongs/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        首页
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" >
      
      <label class="md-nav__link" for="nav-2">
        数据分析笔记
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="数据分析笔记" data-md-level="1">
        <label class="md-nav__title" for="nav-2">
          <span class="md-nav__icon md-icon"></span>
          数据分析笔记
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/main/" class="md-nav__link">
        第零章 开始之前
      </a>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-2" type="checkbox" id="nav-2-2" >
      
      <label class="md-nav__link" for="nav-2-2">
        第一章 常用工具
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第一章 常用工具" data-md-level="2">
        <label class="md-nav__title" for="nav-2-2">
          <span class="md-nav__icon md-icon"></span>
          第一章 常用工具
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C01_Common-Utility/01_database/" class="md-nav__link">
        01. 基因组公共数据库
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C01_Common-Utility/02_get-data/" class="md-nav__link">
        02. 下载基因组数据
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C01_Common-Utility/03_entrez/" class="md-nav__link">
        03. Entrez 工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C01_Common-Utility/04_emboss/" class="md-nav__link">
        04. EMBOSS 工具集
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C01_Common-Utility/05_blast/" class="md-nav__link">
        05. Blast 比对
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C01_Common-Utility/06_fastqc/" class="md-nav__link">
        06. FastQC
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C01_Common-Utility/07_multiqc/" class="md-nav__link">
        07. Multiqc
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C01_Common-Utility/08_seqkit/" class="md-nav__link">
        08. SeqKit 工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C01_Common-Utility/09_kmer/" class="md-nav__link">
        09. Kmer 工具
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-3" type="checkbox" id="nav-2-3" >
      
      <label class="md-nav__link" for="nav-2-3">
        第二章 组装与比对
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第二章 组装与比对" data-md-level="2">
        <label class="md-nav__title" for="nav-2-3">
          <span class="md-nav__icon md-icon"></span>
          第二章 组装与比对
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C02_Raw-Data-to-Genome/01_bwa-samtools-pipeline/" class="md-nav__link">
        01. 基本比对流程
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C02_Raw-Data-to-Genome/02_quast/" class="md-nav__link">
        02. Quast 验证
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C02_Raw-Data-to-Genome/03_bacteria-assembly/" class="md-nav__link">
        03. 细菌基因组组装
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C02_Raw-Data-to-Genome/04_virus-assembly/" class="md-nav__link">
        04. 病毒基因组组装
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C02_Raw-Data-to-Genome/05_gatk3/" class="md-nav__link">
        05. GATK3
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C02_Raw-Data-to-Genome/06_gatk4/" class="md-nav__link">
        06. GATK4
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C02_Raw-Data-to-Genome/11_long-noisy-reads-assembly/" class="md-nav__link">
        11. 长序列比对
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C02_Raw-Data-to-Genome/12_hybrid-assembly/" class="md-nav__link">
        12. 二代三代混合组装
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C02_Raw-Data-to-Genome/14_bwa-long-sequence-mapping/" class="md-nav__link">
        14. bwa 长序列比对
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-4" type="checkbox" id="nav-2-4" >
      
      <label class="md-nav__link" for="nav-2-4">
        第三章 基因组注释
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第三章 基因组注释" data-md-level="2">
        <label class="md-nav__title" for="nav-2-4">
          <span class="md-nav__icon md-icon"></span>
          第三章 基因组注释
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C03_Genome-Annotation/01_prokka/" class="md-nav__link">
        01. Prokka 快速注释
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C03_Genome-Annotation/02_rast/" class="md-nav__link">
        02. 在线注释工具 rast
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C03_Genome-Annotation/03_pgaap/" class="md-nav__link">
        03. 注释工具 PGAAP
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C03_Genome-Annotation/04_snpeff/" class="md-nav__link">
        04. SNP注释工具 snpEFF
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C03_Genome-Annotation/05_cog/" class="md-nav__link">
        05. COG注释
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-5" type="checkbox" id="nav-2-5" >
      
      <label class="md-nav__link" for="nav-2-5">
        第四章 基因组比较
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第四章 基因组比较" data-md-level="2">
        <label class="md-nav__title" for="nav-2-5">
          <span class="md-nav__icon md-icon"></span>
          第四章 基因组比较
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C06_Genome-Compare/01_mummer/" class="md-nav__link">
        01. mummer
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C06_Genome-Compare/02_nucdiff/" class="md-nav__link">
        02. nucdiff
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C06_Genome-Compare/03_mauve/" class="md-nav__link">
        03. mauve
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C06_Genome-Compare/04_sibelia/" class="md-nav__link">
        04. sibelia
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C06_Genome-Compare/05_ani/" class="md-nav__link">
        05. ani
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C06_Genome-Compare/10_bigsi/" class="md-nav__link">
        10. bigsi
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-6" type="checkbox" id="nav-2-6" >
      
      <label class="md-nav__link" for="nav-2-6">
        第五章 泛基因组分析
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第五章 泛基因组分析" data-md-level="2">
        <label class="md-nav__title" for="nav-2-6">
          <span class="md-nav__icon md-icon"></span>
          第五章 泛基因组分析
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C07_PanGenome-and-CoreGenome/01_pangenome/" class="md-nav__link">
        01. Pangenome 概述
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C07_PanGenome-and-CoreGenome/02_get_homologues/" class="md-nav__link">
        02. Get_Homologues 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C07_PanGenome-and-CoreGenome/03_bpga/" class="md-nav__link">
        03. BPGA 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C07_PanGenome-and-CoreGenome/04_roary/" class="md-nav__link">
        04. Roary 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C07_PanGenome-and-CoreGenome/05_panseq/" class="md-nav__link">
        05. panseq 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C07_PanGenome-and-CoreGenome/07_orthmcl/" class="md-nav__link">
        07. OrthMCL 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C07_PanGenome-and-CoreGenome/08_itep/" class="md-nav__link">
        08. ITEP 分析
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-7" type="checkbox" id="nav-2-7" >
      
      <label class="md-nav__link" for="nav-2-7">
        第六章 基因组可视化
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第六章 基因组可视化" data-md-level="2">
        <label class="md-nav__title" for="nav-2-7">
          <span class="md-nav__icon md-icon"></span>
          第六章 基因组可视化
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C08_Genome-Visualization/01_igv/" class="md-nav__link">
        01. igv 查看比对结果
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C08_Genome-Visualization/02_brig/" class="md-nav__link">
        02. BRIG 分析工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C08_Genome-Visualization/03_circos/" class="md-nav__link">
        03. circos 工具使用
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-8" type="checkbox" id="nav-2-8" >
      
      <label class="md-nav__link" for="nav-2-8">
        第七章 综合分析平台
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第七章 综合分析平台" data-md-level="2">
        <label class="md-nav__title" for="nav-2-8">
          <span class="md-nav__icon md-icon"></span>
          第七章 综合分析平台
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C09_Analysis-Platform/01_docker/" class="md-nav__link">
        01. 基于 docker 镜像
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C09_Analysis-Platform/02_galaxy/" class="md-nav__link">
        02. Galaxy 平台
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C09_Analysis-Platform/03_amazon/" class="md-nav__link">
        03. Amazon EC2 云平台
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C09_Analysis-Platform/04_others/" class="md-nav__link">
        04. 其他平台
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C16_Standalone-Platform/01_anvio/" class="md-nav__link">
        05. anvio
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C16_Standalone-Platform/02_galaxy/" class="md-nav__link">
        02. galaxy
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-9" type="checkbox" id="nav-2-9" >
      
      <label class="md-nav__link" for="nav-2-9">
        第八章 重组与种群分析
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第八章 重组与种群分析" data-md-level="2">
        <label class="md-nav__title" for="nav-2-9">
          <span class="md-nav__icon md-icon"></span>
          第八章 重组与种群分析
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C10_Recombination-and-Population-Structure/01_virus-recombination/" class="md-nav__link">
        01. 病毒重组
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C10_Recombination-and-Population-Structure/02_gubbins/" class="md-nav__link">
        02. Gubbins 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C10_Recombination-and-Population-Structure/03_baps/" class="md-nav__link">
        03. BAPS
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-10" type="checkbox" id="nav-2-10" >
      
      <label class="md-nav__link" for="nav-2-10">
        第九章 细菌扫描工具
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第九章 细菌扫描工具" data-md-level="2">
        <label class="md-nav__title" for="nav-2-10">
          <span class="md-nav__icon md-icon"></span>
          第九章 细菌扫描工具
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C11_Insilico-Scanning/01_srst2/" class="md-nav__link">
        01. SRST2 短序列扫描工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C11_Insilico-Scanning/02_mlst/" class="md-nav__link">
        02. MLST 扫描工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C11_Insilico-Scanning/03_ariba/" class="md-nav__link">
        03. Ariba 耐药基因
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C11_Insilico-Scanning/04_abricate/" class="md-nav__link">
        04. Abricate 多功能扫描工具
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-11" type="checkbox" id="nav-2-11" >
      
      <label class="md-nav__link" for="nav-2-11">
        第十章 宏基因组分析
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十章 宏基因组分析" data-md-level="2">
        <label class="md-nav__title" for="nav-2-11">
          <span class="md-nav__icon md-icon"></span>
          第十章 宏基因组分析
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C12_Metagenomics-Analysis/01_kraken/" class="md-nav__link">
        01. kraken
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C12_Metagenomics-Analysis/02_centrifuge/" class="md-nav__link">
        02. centrifuge
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C12_Metagenomics-Analysis/03_bracken/" class="md-nav__link">
        03. bracken
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C12_Metagenomics-Analysis/04_sourmash/" class="md-nav__link">
        04. sourmash
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C12_Metagenomics-Analysis/05_clark/" class="md-nav__link">
        05. clark
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C12_Metagenomics-Analysis/06_krona/" class="md-nav__link">
        06. krona
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C12_Metagenomics-Analysis/07_pavian/" class="md-nav__link">
        07. pavian
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C12_Metagenomics-Analysis/08_supri/" class="md-nav__link">
        08. supri
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C12_Metagenomics-Analysis/09_shotgun/" class="md-nav__link">
        09. shotgun 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C12_Metagenomics-Analysis/10_16s/" class="md-nav__link">
        10. 16S 种群分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C12_Metagenomics-Analysis/11_kmerfinder/" class="md-nav__link">
        11. KmerFinder
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C12_Metagenomics-Analysis/12_meta-beetl/" class="md-nav__link">
        12. Meta-BEETL
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-12" type="checkbox" id="nav-2-12" >
      
      <label class="md-nav__link" for="nav-2-12">
        第十一章 单分子测序
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十一章 单分子测序" data-md-level="2">
        <label class="md-nav__title" for="nav-2-12">
          <span class="md-nav__icon md-icon"></span>
          第十一章 单分子测序
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C13_Single-Molecular-Sequencing/01_nanopore/" class="md-nav__link">
        01. nanopore
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C13_Single-Molecular-Sequencing/02_pacbio/" class="md-nav__link">
        02. pacbio
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C13_Single-Molecular-Sequencing/03_canu/" class="md-nav__link">
        03. canu
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C13_Single-Molecular-Sequencing/04_seqs-polish/" class="md-nav__link">
        04. trimming
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C13_Single-Molecular-Sequencing/06_hybrid-assembling/" class="md-nav__link">
        06. hybrid
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-13" type="checkbox" id="nav-2-13" >
      
      <label class="md-nav__link" for="nav-2-13">
        第十二章 多重序列比对
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十二章 多重序列比对" data-md-level="2">
        <label class="md-nav__title" for="nav-2-13">
          <span class="md-nav__icon md-icon"></span>
          第十二章 多重序列比对
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C14_Multiple-Sequence-Alignment/01_msa/" class="md-nav__link">
        01. 多重序列比对
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-14" type="checkbox" id="nav-2-14" >
      
      <label class="md-nav__link" for="nav-2-14">
        第十三章 进化树构建分析
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十三章 进化树构建分析" data-md-level="2">
        <label class="md-nav__title" for="nav-2-14">
          <span class="md-nav__icon md-icon"></span>
          第十三章 进化树构建分析
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C15_Phylogeny-Analysis/00_all-about-phylogeny/" class="md-nav__link">
        00. 构建进化书
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C15_Phylogeny-Analysis/02_raxml.md" class="md-nav__link">
        01. RAxML
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C15_Phylogeny-Analysis/01_figtree.md" class="md-nav__link">
        02. figtree
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C15_Phylogeny-Analysis/02_ggtree.md" class="md-nav__link">
        03. ggtree
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-15" type="checkbox" id="nav-2-15" >
      
      <label class="md-nav__link" for="nav-2-15">
        第十七章 分子地理流行病学
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十七章 分子地理流行病学" data-md-level="2">
        <label class="md-nav__title" for="nav-2-15">
          <span class="md-nav__icon md-icon"></span>
          第十七章 分子地理流行病学
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C17_Pathogen-Geospatial-Analysis/04_cartodb/" class="md-nav__link">
        04. CartoDB
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C17_Pathogen-Geospatial-Analysis/05_leaflet/" class="md-nav__link">
        05. leaflet
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-16" type="checkbox" id="nav-2-16" >
      
      <label class="md-nav__link" for="nav-2-16">
        第十八章 分析实践
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十八章 分析实践" data-md-level="2">
        <label class="md-nav__title" for="nav-2-16">
          <span class="md-nav__icon md-icon"></span>
          第十八章 分析实践
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C18_Analysis-Examples/01_trace-Pae-from-different-source-by-SNP/" class="md-nav__link">
        01. 细菌SNP溯源分析一
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C18_Analysis-Examples/02_comparative-analysis-Ban-genomes/" class="md-nav__link">
        02. 细菌比较基因组学研究
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-17" type="checkbox" id="nav-2-17" >
      
      <label class="md-nav__link" for="nav-2-17">
        第十九章 工具流工具
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十九章 工具流工具" data-md-level="2">
        <label class="md-nav__title" for="nav-2-17">
          <span class="md-nav__icon md-icon"></span>
          第十九章 工具流工具
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C19_Pipelines-and-Workflows/05_workflow-software/" class="md-nav__link">
        05. workflow工具软件
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-18" type="checkbox" id="nav-2-18" >
      
      <label class="md-nav__link" for="nav-2-18">
        第二十章 其他
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第二十章 其他" data-md-level="2">
        <label class="md-nav__title" for="nav-2-18">
          <span class="md-nav__icon md-icon"></span>
          第二十章 其他
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../book/C20_Others/02_gi-to-accession/" class="md-nav__link">
        02. GI转Accession
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" >
      
      <label class="md-nav__link" for="nav-3">
        专项教程
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="专项教程" data-md-level="1">
        <label class="md-nav__title" for="nav-3">
          <span class="md-nav__icon md-icon"></span>
          专项教程
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-3-1" type="checkbox" id="nav-3-1" >
      
      <label class="md-nav__link" for="nav-3-1">
        Linux入门
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Linux入门" data-md-level="2">
        <label class="md-nav__title" for="nav-3-1">
          <span class="md-nav__icon md-icon"></span>
          Linux入门
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Linux/01_linux-installation/" class="md-nav__link">
        01. Linux 安装示例
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Linux/02_basic-cml/" class="md-nav__link">
        02. Linux 基本命令
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Linux/03_linux-software/" class="md-nav__link">
        03. Linux 软件管理
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Linux/04_awk/" class="md-nav__link">
        04. awk 列处理工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Linux/05_sed/" class="md-nav__link">
        05. sed 行处理工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Linux/06_tools/" class="md-nav__link">
        06. 其他命令行工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Linux/07_vim/" class="md-nav__link">
        07. vim
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-3-2" type="checkbox" id="nav-3-2" >
      
      <label class="md-nav__link" for="nav-3-2">
        常用编程语言
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="常用编程语言" data-md-level="2">
        <label class="md-nav__title" for="nav-3-2">
          <span class="md-nav__icon md-icon"></span>
          常用编程语言
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Programming/01_shell/" class="md-nav__link">
        01. Shell 脚本
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Programming/02_perl/" class="md-nav__link">
        02. Perl 编程
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Programming/03_python/" class="md-nav__link">
        03. Python 编程
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Programming/04_biopython/" class="md-nav__link">
        04. Biopython 编程
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Programming/05_r/" class="md-nav__link">
        05. R 编程
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-3-3" type="checkbox" id="nav-3-3" >
      
      <label class="md-nav__link" for="nav-3-3">
        数据文件格式
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="数据文件格式" data-md-level="2">
        <label class="md-nav__title" for="nav-3-3">
          <span class="md-nav__icon md-icon"></span>
          数据文件格式
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Format/01_fasta/" class="md-nav__link">
        01. fasta
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Format/02_embl/" class="md-nav__link">
        02. embl
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Format/05_fastq.md" class="md-nav__link">
        03. fastq
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Format/04_gcg/" class="md-nav__link">
        04. gcg
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Format/05_sam/" class="md-nav__link">
        05. sam
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Format/06_bam/" class="md-nav__link">
        06. bam
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Format/07_bed/" class="md-nav__link">
        07. bed
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Format/08_vcf/" class="md-nav__link">
        08. vcf
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Format/09_gff/" class="md-nav__link">
        09. gff
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/Format/10_cram/" class="md-nav__link">
        10. cram
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4" >
      
      <label class="md-nav__link" for="nav-4">
        实验手册
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="实验手册" data-md-level="1">
        <label class="md-nav__title" for="nav-4">
          <span class="md-nav__icon md-icon"></span>
          实验手册
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../protocol/" class="md-nav__link">
        
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../protocol/artic-network-covid19/" class="md-nav__link">
        ArticNetwork Covid-19测序
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" >
      
      <label class="md-nav__link" for="nav-5">
        生信笔记
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="生信笔记" data-md-level="1">
        <label class="md-nav__title" for="nav-5">
          <span class="md-nav__icon md-icon"></span>
          生信笔记
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../my-vim-ide/" class="md-nav__link">
        01 TEST
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../blast2sra/" class="md-nav__link">
        02 blast2SRA
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../tes/" class="md-nav__link">
        03 Rmarkdown
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../test0316/" class="md-nav__link">
        04 SPAdes
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../KEGG/" class="md-nav__link">
        05 KEGG注释
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6" >
      
      <label class="md-nav__link" for="nav-6">
        关于我们
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="关于我们" data-md-level="1">
        <label class="md-nav__title" for="nav-6">
          <span class="md-nav__icon md-icon"></span>
          关于我们
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../about/" class="md-nav__link">
        
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-maker" class="md-nav__link">
    1 用maker进行真核基因组的基因预测
  </a>
  
    <nav class="md-nav" aria-label="1 用maker进行真核基因组的基因预测">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-maker" class="md-nav__link">
    1.1 获取maker的配置文件
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#_2" class="md-nav__link">
    配置文件
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#12-maker" class="md-nav__link">
    1.2 准备maker所需要的预测基因数据库
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#13-maker" class="md-nav__link">
    1.3 运行maker
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" class="md-nav__link">
    2 整合其它软件获得的注释结果
  </a>
  
    <nav class="md-nav" aria-label="2 整合其它软件获得的注释结果">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21" class="md-nav__link">
    2.1 软件配置
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22" class="md-nav__link">
    2.2 原始数据
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#23" class="md-nav__link">
    2.3 注释重复序列
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-maker" class="md-nav__link">
    3 调整maker注释参数
  </a>
  
    <nav class="md-nav" aria-label="3 调整maker注释参数">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31" class="md-nav__link">
    3.1 第一轮注释
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32" class="md-nav__link">
    3.2 第二轮注释
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#33" class="md-nav__link">
    3.3 第三轮注释
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#34" class="md-nav__link">
    3.4 迭代和整理结果
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/indexofire/pathongs/edit/master/docs/blog/test0317.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                <h1 id="_1">真核生物基因组的基因预测和注释<a class="headerlink" href="#_1" title="Permanent link">&para;</a></h1>
<h2 id="1-maker">1 用maker进行真核基因组的基因预测<a class="headerlink" href="#1-maker" title="Permanent link">&para;</a></h2>
<blockquote>
<p>maker结合了重复序列屏蔽，augustus，genemark，snap从头预测，同源注释以及RNA-seq的方法，其本身并不属于一种预测基因的方法，其为一套整合各个方法的流程。</p>
</blockquote>
<h3 id="11-maker">1.1 获取maker的配置文件<a class="headerlink" href="#11-maker" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="n">maker</span> <span class="o">-</span><span class="n">CTL</span>
</code></pre></div>
<h3 id="_2">配置文件<a class="headerlink" href="#_2" title="Permanent link">&para;</a></h3>
<ul>
<li>
<p>maker_bopts.ctl 可不用修改</p>
</li>
<li>
<p>maker_exe.ctl 执行路径</p>
</li>
<li>
<p>maker_opts.ctl 重点修改对象</p>
</li>
</ul>
<h3 id="12-maker">1.2 准备maker所需要的预测基因数据库<a class="headerlink" href="#12-maker" title="Permanent link">&para;</a></h3>
<ul>
<li>genome：fasta格式，由上一步组装获得</li>
<li>EST：EST 原指表达序列标签(expressedsequencetags, *EST*s)，是指从不同组织来源的cDNA序列，在当下可以用组装的转录本代替。</li>
<li>altest：同源物种的EST，fasta格式</li>
<li>protein：同源物种的蛋白序列，fasta格式</li>
</ul>
<p>组装的转录本可以由<a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki">Trinity</a>获得。</p>
<p>Trinity是<a href="https://links.jianshu.com/go?to=http%3A%2F%2Fwww.broadinstitute.org%2F">Broad Institute</a>和<a href="https://links.jianshu.com/go?to=http%3A%2F%2Fwww.cs.huji.ac.il%2F">Hebrew University of Jerusalem</a>开发的RNA-Seq数据 转录组组装工具，包括三个模块，</p>
<ul>
<li>Inchworn(尺蠖): 将RNA-seq数据组装成单个转录本，通常是主要转录亚型的全长转录本</li>
<li>Chrysalis(蛹): 这一步将上一步得到contig进行聚类，对于每个聚类构建完整的德布罗意图(*de Bruijin*graph)。每个转录本表示的是给定基因或者一组有着共同序列的基因的全部转录组成。 之后会根据图中不相交的点对全部短读数据进行拆分</li>
<li>Butterfly(蝴蝶): 并行处理各个图(graph), 追踪每个图中的短读和配对短读的路径，最后报告可变剪切亚型的全长转录本，并且区分出旁系同源基因的转录本</li>
</ul>
<p><img alt="Trinity" src="https://upload-images.jianshu.io/upload_images/2013053-e88fc3f8d879519f.png" /></p>
<p>其输入结果和输出结果都十分简单。</p>
<div class="highlight"><pre><span></span><code>Trinity --seqType fq --max_memory 50G  <span class="se">\</span>
         --left condA_1.fq.gz,condB_1.fq.gz,condC_1.fq.gz <span class="se">\</span>
         --right condA_2.fq.gz,condB_2.fq.gz,condC_2.fq.gz <span class="se">\</span>
         --CPU <span class="m">6</span>  --min_kmer_cov <span class="m">2</span> <span class="se">\</span>
         --jaccard_clip
</code></pre></div>
<p>为加快速度，可以用--normalize_max_read_cov减少高表达的基因的reads数，用<code>--min_kmer_cov 2</code>参数丢弃uniquely occurring kmer, 从而降低内存消耗,也可以用fastp去除低质量的reads，用cd-hit-est来去除重复的reads。</p>
<p>对于真菌转录组数据，由于基因密度较高，需添加--jaccard_clip</p>
<p>最终结果为Trinity.fasta文件</p>
<h3 id="13-maker">1.3 运行maker<a class="headerlink" href="#13-maker" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>maker &amp; maker.log &amp;
</code></pre></div>
<h2 id="2">2 整合其它软件获得的注释结果<a class="headerlink" href="#2" title="Permanent link">&para;</a></h2>
<h3 id="21">2.1 软件配置<a class="headerlink" href="#21" title="Permanent link">&para;</a></h3>
<ol>
<li><a href="http://www.repeatmasker.org/RepeatModeler/">RepeatModeler</a> and <a href="http://www.repeatmasker.org/RMDownload.html">RepeatMasker</a> with all dependencies (I used NCBI BLAST) and <a href="http://www.girinst.org/repbase/">RepBase</a> </li>
<li>MAKER</li>
<li><a href="http://bioinf.uni-greifswald.de/augustus/">Augustus</a> </li>
<li><a href="http://busco.ezlab.org/">BUSCO</a> </li>
<li><a href="http://korflab.ucdavis.edu/software.html">SNAP</a> </li>
<li><a href="https://bedtools.readthedocs.io/en/latest/">BEDtools</a></li>
</ol>
<h3 id="22">2.2 原始数据<a class="headerlink" href="#22" title="Permanent link">&para;</a></h3>
<ol>
<li>组装好的基因组文件 ：genome.fasta</li>
<li>组装好的转录本文件：trinity.fasta</li>
<li>同源的蛋白质序列：protein.fasta (from <a href="https://www.ncbi.nlm.nih.gov/genome">NCB</a> or <a href="www.uniprot.org">uniport</a> or <a href="https://genome.jgi.doe.gov/portal">JGI</a> )</li>
<li>repeat library (手动注释或者同源物种)</li>
</ol>
<h3 id="23">2.3 注释重复序列<a class="headerlink" href="#23" title="Permanent link">&para;</a></h3>
<ol>
<li><em>De Novo</em> Repeat Identification</li>
</ol>
<p>对于真核生物来说，注释其重复序列对于后期基因组的注释是非常重要的。虽然有很多重复序列的数据库如Repbase，但从头对其基因组进行重复序列注释依然是十分有必要的。</p>
<p>通过RepeatModeler，单核单线程注释几十兆的小型基因组只需要几个小时。</p>
<div class="highlight"><pre><span></span><code>BuildDatabase -name <span class="nb">test</span> -engine ncbi genome.fasta
RepeatModeler -pa <span class="m">1</span> -engine ncbi -database <span class="nb">test</span> 
</code></pre></div>
<p>得到两个结果文件：</p>
<ul>
<li>test-families.fa  : Consensus sequence</li>
<li>
<p>test-families.stk : Seed alignments</p>
</li>
<li>
<p>Full Repeat Annotation</p>
</li>
</ul>
<p>通过RepeatMasker，可以进行更加深入的重复序列注释。我们需要用到上一步的test-families.fa文件</p>
<div class="highlight"><pre><span></span><code>RepeatMasker -pa <span class="m">1</span> -e ncbi -lib test-families.fa -dir mask01/ genome.fasta
</code></pre></div>
<p>可以获得4个文件：</p>
<ul>
<li>test.fasta.tbl</li>
<li>test.fasta.cat.gz</li>
<li>test.fasta.masked</li>
<li>test.fasta.out</li>
</ul>
<p>针对这4个文件，我们可以结合Repbase数据库，进行第二轮重复序列注释</p>
<div class="highlight"><pre><span></span><code>RepeatMasker -pa <span class="m">1</span> -e ncbi -species fungi -dir mask02 test.fasta.masked
</code></pre></div>
<p>同样可以获得4个文件</p>
<ul>
<li>test.fasta.out.out</li>
<li>test.fasta.masked.masked</li>
<li>test.fasta.masked.cat.gz</li>
<li>test.fasta.masked.tbl</li>
</ul>
<p>通过一些脚本，让这些注释文件合并，用于输入给maker</p>
<div class="highlight"><pre><span></span><code>cp test.fasta.masked.masked full_mask/test.full_mask.fa
cp test.fasta.out.out full_mask/test.full_mask.out
gunzip test.fasta.cat.gz test.fasta.masked.cat.gz 
cat *.cat &gt; full_mask/test.full_mask.cat
<span class="nb">cd</span> full_mask
ProcessRepeats -species fungi test.full_mask.cat
rmOutToGFF3.pl test.full_mask.out &gt; test.full_mask.gff3 <span class="c1">#generate gff file</span>
grep -v -e <span class="s2">&quot;Satellite&quot;</span> -e <span class="s2">&quot;)n&quot;</span> -e <span class="s2">&quot;-rich&quot;</span> test.full_mask.gff3 &gt; test.full_mask.out.complex.gff3 <span class="c1"># isolate complex repeats</span>
cat test.full_mask.complex.gff3 <span class="p">|</span> perl -ane <span class="s1">&#39;$id; if(!/^\#/){@F = split(/\t/, $_); chomp $F[-1];$id++; $F[-1] .= &quot;\;ID=$id&quot;; $_ = join(&quot;\t&quot;, @F).&quot;\n&quot;} print $_&#39;</span> &gt; test.full_mask.complex.reformat.gff3
</code></pre></div>
<p>最后生成的gff3文件可以直接用于maker的注释。</p>
<h2 id="3-maker">3 调整maker注释参数<a class="headerlink" href="#3-maker" title="Permanent link">&para;</a></h2>
<p>初始化maker</p>
<h3 id="31">3.1 第一轮注释<a class="headerlink" href="#31" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>maker -CTL
cp maker_opt.ctl round1_maker_opt.ctl
vim round1_maker_opt.ctl
<span class="c1">#-----Genome (these are always required)</span>
<span class="nv">genome</span><span class="o">=</span>/home/castoelab/Desktop/daren/boa_annotation/test.fasta <span class="c1">#genome sequence (fasta file or fasta embeded in GFF3 file)</span>
<span class="nv">organism_type</span><span class="o">=</span>eukaryotic <span class="c1">#eukaryotic or prokaryotic. Default is eukaryotic默认是真核</span>

<span class="c1">#-----Re-annotation Using MAKER Derived GFF3</span>
<span class="nv">maker_gff</span><span class="o">=</span>test.full_mask.complex.reformat.gff3 <span class="c1">#MAKER derived GFF3 file第一轮注释通常没有gff3文件</span>
<span class="nv">est_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use ESTs in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">altest_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use alternate organism ESTs in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">protein_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use protein alignments in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">rm_pass</span><span class="o">=</span><span class="m">1</span> <span class="c1">#use repeats in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">model_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use gene models in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">pred_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use ab-initio predictions in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">other_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#passthrough anyything else in maker_gff: 1 = yes, 0 = no</span>

<span class="c1">#-----EST Evidence (for best results provide a file for at least one)</span>
<span class="nv">est</span><span class="o">=</span>/home/castoelab/Desktop/daren/boa_annotation/trinity.fasta <span class="c1">#set of ESTs or assembled mRNA-seq in fasta format转录组的组装结果</span>
<span class="nv">altest</span><span class="o">=</span>/home/castoelab/Desktop/daren/boa_annotation/est/test.est.fasta <span class="c1">#EST/cDNA sequence file in fasta format from an alternate organism近源物种的转录组组装结果</span>
<span class="nv">est_gff</span><span class="o">=</span> <span class="c1">#aligned ESTs or mRNA-seq from an external GFF3 file</span>
<span class="nv">altest_gff</span><span class="o">=</span> <span class="c1">#aligned ESTs from a closly relate species in GFF3 format</span>

<span class="c1">#-----Protein Homology Evidence (for best results provide a file for at least one)</span>
<span class="nv">protein</span><span class="o">=</span>/home/castoelab/Desktop/daren/boa_annotation/protein_files_other_squamates/test.fasta  <span class="c1">#protein sequence file in fasta format (i.e. from mutiple oransisms)</span>
<span class="nv">protein_gff</span><span class="o">=</span>  <span class="c1">#aligned protein homology evidence from an external GFF3 file</span>

<span class="c1">#-----Repeat Masking (leave values blank to skip repeat masking)</span>
<span class="nv">model_org</span><span class="o">=</span>simple <span class="c1">#select a model organism for RepBase masking in RepeatMasker</span>
<span class="nv">rmlib</span><span class="o">=</span> <span class="c1">#provide an organism specific repeat library in fasta format for RepeatMasker</span>
<span class="nv">repeat_protein</span><span class="o">=</span>/opt/maker/data/te_proteins.fasta <span class="c1">#provide a fasta file of transposable element proteins for RepeatRunner</span>
<span class="nv">rm_gff</span><span class="o">=</span>/home/castoelab/Desktop/daren/boa_annotation/full_mask/test.full_mask.complex.reformat.gff3 <span class="c1">#pre-identified repeat elements from an external GFF3 file</span>
<span class="nv">prok_rm</span><span class="o">=</span><span class="m">0</span> <span class="c1">#forces MAKER to repeatmask prokaryotes (no reason to change this), 1 = yes, 0 = no</span>
<span class="nv">softmask</span><span class="o">=</span><span class="m">1</span> <span class="c1">#use soft-masking rather than hard-masking in BLAST (i.e. seg and dust filtering)</span>

<span class="c1">#-----Gene Prediction</span>
<span class="nv">snaphmm</span><span class="o">=</span> <span class="c1">#SNAP HMM file</span>
<span class="nv">gmhmm</span><span class="o">=</span> <span class="c1">#GeneMark HMM file</span>
<span class="nv">augustus_species</span><span class="o">=</span> <span class="c1">#Augustus gene prediction species model</span>
<span class="nv">fgenesh_par_file</span><span class="o">=</span> <span class="c1">#FGENESH parameter file</span>
<span class="nv">pred_gff</span><span class="o">=</span> <span class="c1">#ab-initio predictions from an external GFF3 file</span>
<span class="nv">model_gff</span><span class="o">=</span> <span class="c1">#annotated gene models from an external GFF3 file (annotation pass-through)</span>
<span class="nv">est2genome</span><span class="o">=</span><span class="m">1</span> <span class="c1">#infer gene predictions directly from ESTs, 1 = yes, 0 = no</span>
<span class="nv">protein2genome</span><span class="o">=</span><span class="m">1</span> <span class="c1">#infer predictions from protein homology, 1 = yes, 0 = no</span>
<span class="nv">trna</span><span class="o">=</span><span class="m">0</span> <span class="c1">#find tRNAs with tRNAscan, 1 = yes, 0 = no</span>
<span class="nv">snoscan_rrna</span><span class="o">=</span> <span class="c1">#rRNA file to have Snoscan find snoRNAs</span>
<span class="nv">unmask</span><span class="o">=</span><span class="m">0</span> <span class="c1">#also run ab-initio prediction programs on unmasked sequence, 1 = yes, 0 = no</span>

<span class="c1">#-----Other Annotation Feature Types (features MAKER doesn&#39;t recognize)</span>
<span class="nv">other_gff</span><span class="o">=</span> <span class="c1">#extra features to pass-through to final MAKER generated GFF3 file</span>

<span class="c1">#-----External Application Behavior Options</span>
<span class="nv">alt_peptide</span><span class="o">=</span>C <span class="c1">#amino acid used to replace non-standard amino acids in BLAST databases</span>
<span class="nv">cpus</span><span class="o">=</span><span class="m">1</span> <span class="c1">#max number of cpus to use in BLAST and RepeatMasker (not for MPI, leave 1 when using MPI)</span>

<span class="c1">#-----MAKER Behavior Options</span>
<span class="nv">max_dna_len</span><span class="o">=</span><span class="m">100000</span> <span class="c1">#length for dividing up contigs into chunks (increases/decreases memory usage)</span>
<span class="nv">min_contig</span><span class="o">=</span><span class="m">1</span> <span class="c1">#skip genome contigs below this length (under 10kb are often useless)</span>

<span class="nv">pred_flank</span><span class="o">=</span><span class="m">200</span> <span class="c1">#flank for extending evidence clusters sent to gene predictors</span>
<span class="nv">pred_stats</span><span class="o">=</span><span class="m">0</span> <span class="c1">#report AED and QI statistics for all predictions as well as models</span>
<span class="nv">AED_threshold</span><span class="o">=</span><span class="m">1</span> <span class="c1">#Maximum Annotation Edit Distance allowed (bound by 0 and 1)</span>
<span class="nv">min_protein</span><span class="o">=</span><span class="m">0</span> <span class="c1">#require at least this many amino acids in predicted proteins</span>
<span class="nv">alt_splice</span><span class="o">=</span><span class="m">0</span> <span class="c1">#Take extra steps to try and find alternative splicing, 1 = yes, 0 = no</span>
<span class="nv">always_complete</span><span class="o">=</span><span class="m">0</span> <span class="c1">#extra steps to force start and stop codons, 1 = yes, 0 = no</span>
<span class="nv">map_forward</span><span class="o">=</span><span class="m">0</span> <span class="c1">#map names and attributes forward from old GFF3 genes, 1 = yes, 0 = no</span>
<span class="nv">keep_preds</span><span class="o">=</span><span class="m">0</span> <span class="c1">#Concordance threshold to add unsupported gene prediction (bound by 0 and 1)</span>

<span class="nv">split_hit</span><span class="o">=</span><span class="m">10000</span> <span class="c1">#length for the splitting of hits (expected max intron size for evidence alignments)</span>
<span class="nv">single_exon</span><span class="o">=</span><span class="m">0</span> <span class="c1">#consider single exon EST evidence when generating annotations, 1 = yes, 0 = no</span>
<span class="nv">single_length</span><span class="o">=</span><span class="m">250</span> <span class="c1">#min length required for single exon ESTs if &#39;single_exon is enabled&#39;</span>
<span class="nv">correct_est_fusion</span><span class="o">=</span><span class="m">0</span> <span class="c1">#limits use of ESTs in annotation to avoid fusion genes</span>

<span class="nv">tries</span><span class="o">=</span><span class="m">2</span> <span class="c1">#number of times to try a contig if there is a failure for some reason</span>
<span class="nv">clean_try</span><span class="o">=</span><span class="m">0</span> <span class="c1">#remove all data from previous run before retrying, 1 = yes, 0 = no</span>
<span class="nv">clean_up</span><span class="o">=</span><span class="m">0</span> <span class="c1">#removes theVoid directory with individual analysis files, 1 = yes, 0 = no</span>
<span class="nv">TMP</span><span class="o">=</span> <span class="c1">#specify a directory other than the system default temporary directory for temporary files</span>
</code></pre></div>
<p>运行maker</p>
<div class="highlight"><pre><span></span><code>maker -base test_rnd1 round1_maker_opts.ctl maker_bopts.ctl maker_exe.ctl
</code></pre></div>
<p>整理第一轮注释的结果</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> test.maker.out
gff3_merge -d test_rnd1_master_datastore_index.log
fasta_merge -d test_rnd1_master_datastore_index.log
gff3_merge -n -s -d test_rnd1_master_datastore_index.log &gt; test_rnd1.all.maker.noseq.gff <span class="c1">#选择不输出sequence</span>
</code></pre></div>
<h3 id="32">3.2 第二轮注释<a class="headerlink" href="#32" title="Permanent link">&para;</a></h3>
<p>第二轮注释可以用SNAP这个软件，它的速度非常快，可以生成基因的hmm模型，需要用到第一轮注释生成的gff文件</p>
<div class="highlight"><pre><span></span><code>mkdir snap
<span class="nb">cd</span> snap
mv ../test_rnd1.all.maker.noseq.gff .
maker2zff test_rnd1.all.maker.noseq.gff
</code></pre></div>
<p>可以得到两个基因组的注释文件，分别是</p>
<ul>
<li>genome.ann</li>
<li>genome.dna</li>
</ul>
<p>参考<a href="https://github.com/KorfLab/SNAP">SNAP</a>的官方文档，可以对其进行训练</p>
<p>安装SNAP</p>
<div class="highlight"><pre><span></span><code>fathom genome.ann genome.dna -gene-stats &gt; test.gene.stats.txt
fathom genome.ann genome.dna -validate &gt; test.validation.txt
fathom genome.ann genome.dna -categorize <span class="m">1000</span> <span class="c1">#将序列分类，这里的1000推荐用基因平均长度的一半，根据基因密度适当调整</span>
fathom uni.ann uni.dna -export <span class="m">1000</span> -plus# 只输出unigene中正链基因
</code></pre></div>
<p>test.gene.stats.txt用于了解基因的一些基本信息，例如正链和负链的分布，外显子的数量。</p>
<p>test.validation.txt用于检查基因结构的完整性。</p>
<p>最终会生成一下这些文件</p>
<ul>
<li>alt.ann, alt.dna (genes with alternative splicing)</li>
<li>err.ann, err.dna (genes that have errors)</li>
<li>olp.ann, olp.dna (genes that overlap other genes)</li>
<li>wrn.ann, wrn.dna (genes with warnings)</li>
<li>uni.ann, uni.dna (single gene per sequence)</li>
</ul>
<p>这里只用第四类的基因，也就是每个序列上只有一个基因。</p>
<p>用fathom uni.ann uni.dna -export 1000 -plus只输出unigene中正链基因，这一步同样会生成四个文件</p>
<ul>
<li>export.aa 每个基因的蛋白序列</li>
<li>export.ann 正链的基因结构</li>
<li>export.dna 正链的DNA</li>
<li>export.tx 每个基因的转录本</li>
</ul>
<p>接着让forge负责预测参数， 由于输出会很多，所以建议创建个文件夹</p>
<div class="highlight"><pre><span></span><code>mkdir params
<span class="nb">cd</span> params
forge ../export.ann ../export.dna
<span class="nb">cd</span> ..
</code></pre></div>
<p>最后是hmm-assembler.pl构建HMM，即基因模型文件, </p>
<div class="highlight"><pre><span></span><code>hmm-assembler.py <span class="nb">test</span> params/ &gt; test.hmm
</code></pre></div>
<p>完成SNAP的模型构建后，修改”maker_opts.ctl”用以增加该文件，并不再用est和protein直接推测基因结构。</p>
<p>重点是调整基因模型和snaphmm=test.hmm est2genome=0 protein2genome=0</p>
<div class="highlight"><pre><span></span><code>vim maker_opt.ctl
<span class="c1">#-----Re-annotation Using MAKER Derived GFF3</span>
<span class="nv">maker_gff</span><span class="o">=</span>round1.gff <span class="c1">#MAKER derived GFF3 file</span>
<span class="nv">est_pass</span><span class="o">=</span><span class="m">1</span> <span class="c1">#use ESTs in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">altest_pass</span><span class="o">=</span><span class="m">1</span> <span class="c1">#use alternate organism ESTs in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">protein_pass</span><span class="o">=</span><span class="m">1</span> <span class="c1">#use protein alignments in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">rm_pass</span><span class="o">=</span><span class="m">1</span> <span class="c1">#use repeats in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">model_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use gene models in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">pred_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use ab-initio predictions in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">other_pass</span><span class="o">=</span><span class="m">1</span> <span class="c1">#passthrough anyything else in maker_gff: 1 = yes, 0 = no</span>

<span class="c1">#-----EST Evidence (for best results provide a file for at least one)</span>
<span class="nv">est</span><span class="o">=</span> <span class="c1">#set of ESTs or assembled mRNA-seq in fasta format</span>
<span class="nv">altest</span><span class="o">=</span> <span class="c1">#EST/cDNA sequence file in fasta format from an alternate organismest_gff= #aligned ESTs or mRNA-seq from an external GFF3 file</span>
<span class="nv">altest_gff</span><span class="o">=</span> <span class="c1">#aligned ESTs from a closly relate species in GFF3 format</span>

<span class="c1">#-----Protein Homology Evidence (for best results provide a file for at least one)</span>
<span class="nv">protein</span><span class="o">=</span>  <span class="c1">#protein sequence file in fasta format (i.e. from mutiple organisms)</span>
<span class="nv">protein_gff</span><span class="o">=</span>  <span class="c1">#aligned protein homology evidence from an external GFF3 file</span>

<span class="c1">#-----Repeat Masking (leave values blank to skip repeat masking)</span>
<span class="nv">model_org</span><span class="o">=</span> <span class="c1">#select a model organism for RepBase masking in RepeatMasker</span>
<span class="nv">rmlib</span><span class="o">=</span> <span class="c1">#provide an organism specific repeat library in fasta format for RepeatMasker</span>
<span class="nv">repeat_protein</span><span class="o">=</span> <span class="c1">#provide a fasta file of transposable element proteins for RepeatRunner</span>
<span class="nv">rm_gff</span><span class="o">=</span> <span class="c1">#pre-identified repeat elements from an external GFF3 file</span>
<span class="nv">prok_rm</span><span class="o">=</span><span class="m">0</span> <span class="c1">#forces MAKER to repeatmask prokaryotes (no reason to change this), 1 = yes, 0 = no</span>
<span class="nv">softmask</span><span class="o">=</span><span class="m">1</span> <span class="c1">#use soft-masking rather than hard-masking in BLAST (i.e. seg and dust filtering)</span>

<span class="c1">#-----Gene Prediction</span>
<span class="nv">snaphmm</span><span class="o">=</span>test.hmm <span class="c1">#SNAP HMM file</span>
<span class="nv">gmhmm</span><span class="o">=</span> <span class="c1">#GeneMark HMM file</span>
<span class="nv">augustus_species</span><span class="o">=</span> <span class="c1">#Augustus gene prediction species model</span>
<span class="c1"># 略过其他参数</span>
<span class="nv">est2genome</span><span class="o">=</span><span class="m">0</span> <span class="c1">#infer gene predictions directly from ESTs, 1 = yes, 0 = no</span>
<span class="nv">protein2genome</span><span class="o">=</span><span class="m">0</span> <span class="c1">#infer predictions from protein homology, 1 = yes, 0 = no</span>
<span class="c1"># 略过其他参数</span>
</code></pre></div>
<h3 id="33">3.3 第三轮注释<a class="headerlink" href="#33" title="Permanent link">&para;</a></h3>
<p>第三轮注释依赖于Augusts</p>
<p>提取第二轮注释结果中基因及其前后1000bp作为训练数据。</p>
<div class="highlight"><pre><span></span><code>awk -v <span class="nv">OFS</span><span class="o">=</span><span class="s2">&quot;\t&quot;</span> <span class="s1">&#39;{ if ($3 == &quot;mRNA&quot;) print $1, $4, $5 }&#39;</span> ../../round1.gff <span class="p">|</span>awk -v <span class="nv">OFS</span><span class="o">=</span><span class="s2">&quot;\t&quot;</span> <span class="s1">&#39;{ if ($2 &lt; 1000) print $1, &quot;0&quot;, $3+1000; else print $1, $2-1000, $3+1000 }&#39;</span> <span class="p">|</span> bedtools getfasta -fi ../../genome.fasta -bed - -fo round1.all.maker.transcripts1000.fasta
</code></pre></div>
<p>用BUSCO可以较为方便的完成对Augusts的训练，需要注意的是，不同版本的BUSCO和Augusts需要搭配使用，且参数有很大不同。为了优化BUSCO的注释，让其结果更适合给Augusts训练，需要添加--long参数</p>
<p>对于非模式物种，后面需要手动修改AUGUSTS的config文件夹，所以要保证有BUSCO和AUGUSTS安装路径的读写权限。</p>
<div class="highlight"><pre><span></span><code>BUSCO.py -i round1.all.maker.transcripts1000.fasta  -o test_rnd1_maker -l fungi_odb10/ -m genome -c <span class="m">8</span> --long -sp aspergillus_nidulans -z --augustus_parameters<span class="o">=</span><span class="s1">&#39;--progress=true&#39;</span> <span class="c1">#-sp物种需要从BUSCO官方列表里面查,这一步用8个线程需要耗费数个小时</span>
</code></pre></div>
<p>完成后可以在retraining_parameters文件夹下找到对应的模型文件。这里不建议改文件名，如果需要更改文件名，同时也要改文件里面对应的名字。</p>
<p>文件名通常为BUSCO_XXXX_123456,在AUGUST的config/species文件夹里面新建对应的文件夹</p>
<div class="highlight"><pre><span></span><code>mkdir <span class="nv">$AUGUSTUS_CONFIG_PATH</span>/species/BUSCO_XXXX_123456
cp BUSCO_XXXX_123456* <span class="nv">$AUGUSTUS_CONFIG_PATH</span>/species/BUSCO_XXXX_123456/
</code></pre></div>
<p>类似的，我们整理第二轮的gff文件为第三轮注释做准备。注意Augusts的参数需要更改物种为物种文件夹名称</p>
<div class="highlight"><pre><span></span><code><span class="c1"># transcript alignments</span>
awk <span class="s1">&#39;{ if ($2 == &quot;est2genome&quot;) print $0 }&#39;</span> test_rnd1.all.maker.noseq.gff &gt; test_rnd1.all.maker.est2genome.gff
<span class="c1"># protein alignments</span>
awk <span class="s1">&#39;{ if ($2 == &quot;protein2genome&quot;) print $0 }&#39;</span> test_rnd1.all.maker.noseq.gff &gt; test_rnd1.all.maker.protein2genome.gff
<span class="c1"># repeat alignments</span>
awk <span class="s1">&#39;{ if ($2 ~ &quot;repeat&quot;) print $0 }&#39;</span> test_rnd1.all.maker.noseq.gff &gt; test_rnd1.all.maker.repeats.gff
</code></pre></div>
<div class="highlight"><pre><span></span><code>vim round3_maker_opts.ctl
<span class="c1">#-----Genome (these are always required)</span>
<span class="nv">genome</span><span class="o">=</span>genome.fasta <span class="c1">#genome sequence (fasta file or fasta embeded in GFF3 file)</span>
<span class="nv">organism_type</span><span class="o">=</span>eukaryotic <span class="c1">#eukaryotic or prokaryotic. Default is eukaryotic</span>

<span class="c1">#-----Re-annotation Using MAKER Derived GFF3</span>
<span class="nv">maker_gff</span><span class="o">=</span> <span class="c1">#MAKER derived GFF3 file</span>
<span class="nv">est_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use ESTs in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">altest_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use alternate organism ESTs in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">protein_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use protein alignments in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">rm_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use repeats in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">model_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use gene models in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">pred_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#use ab-initio predictions in maker_gff: 1 = yes, 0 = no</span>
<span class="nv">other_pass</span><span class="o">=</span><span class="m">0</span> <span class="c1">#passthrough anyything else in maker_gff: 1 = yes, 0 = no</span>

<span class="c1">#-----EST Evidence (for best results provide a file for at least one)</span>
<span class="nv">est</span><span class="o">=</span> <span class="c1">#set of ESTs or assembled mRNA-seq in fasta format</span>
<span class="nv">altest</span><span class="o">=</span> <span class="c1">#EST/cDNA sequence file in fasta format from an alternate organism</span>
<span class="nv">est_gff</span><span class="o">=</span>test_rnd1.all.maker.est2genome.gff <span class="c1">#aligned ESTs or mRNA-seq from an external GFF3 file</span>
<span class="nv">altest_gff</span><span class="o">=</span> <span class="c1">#aligned ESTs from a closly relate species in GFF3 format</span>

<span class="c1">#-----Protein Homology Evidence (for best results provide a file for at least one)</span>
<span class="nv">protein</span><span class="o">=</span> <span class="c1">#protein sequence file in fasta format (i.e. from mutiple organisms)</span>
<span class="nv">protein_gff</span><span class="o">=</span>test_rnd1.all.maker.protein2genome.gff  <span class="c1">#aligned protein homology evidence from an external GFF3 file</span>


<span class="c1">#-----Repeat Masking (leave values blank to skip repeat masking)</span>
<span class="nv">model_org</span><span class="o">=</span> <span class="c1">#select a model organism for RepBase masking in RepeatMasker</span>
<span class="nv">rmlib</span><span class="o">=</span> <span class="c1">#provide an organism specific repeat library in fasta format for RepeatMasker</span>
<span class="nv">repeat_protein</span><span class="o">=</span> <span class="c1">#o /public/home/software/opt/bio/software/maker/3.01.02-beta/data/te_proteins.fasta #provide a fasta file of transposable element proteins for RepeatRunner</span>
<span class="nv">rm_gff</span><span class="o">=</span>test_rnd1.all.maker.repeats.gff <span class="c1">#pre-identified repeat elements from an external GFF3 file</span>
<span class="nv">prok_rm</span><span class="o">=</span><span class="m">0</span> <span class="c1">#forces MAKER to repeatmask prokaryotes (no reason to change this), 1 = yes, 0 = no</span>
<span class="nv">softmask</span><span class="o">=</span><span class="m">1</span> <span class="c1">#use soft-masking rather than hard-masking in BLAST (i.e. seg and dust filtering)</span>

<span class="c1">#-----Gene Prediction</span>
<span class="nv">snaphmm</span><span class="o">=</span>test.hmm <span class="c1">#SNAP HMM file</span>
<span class="nv">gmhmm</span><span class="o">=</span> <span class="c1">#GeneMark HMM file</span>
<span class="nv">augustus_species</span><span class="o">=</span>BUSCO_xxxxx_123456 <span class="c1">#Augustus gene prediction species model</span>
<span class="nv">fgenesh_par_file</span><span class="o">=</span> <span class="c1">#FGENESH parameter file</span>
<span class="nv">pred_gff</span><span class="o">=</span> <span class="c1">#ab-initio predictions from an external GFF3 file</span>
<span class="nv">model_gff</span><span class="o">=</span> <span class="c1">#annotated gene models from an external GFF3 file (annotation pass-through)</span>
<span class="nv">run_evm</span><span class="o">=</span><span class="m">0</span> <span class="c1">#run EvidencModeler, 1 = yes, 0 = no</span>
<span class="nv">est2genome</span><span class="o">=</span><span class="m">0</span> <span class="c1">#infer gene predictions directly from ESTs, 1 = yes, 0 = no</span>
<span class="nv">protein2genome</span><span class="o">=</span><span class="m">0</span> <span class="c1">#infer predictions from protein homology, 1 = yes, 0 = no</span>
<span class="nv">trna</span><span class="o">=</span><span class="m">0</span> <span class="c1">#find tRNAs with tRNAscan, 1 = yes, 0 = no</span>
<span class="nv">snoscan_rrna</span><span class="o">=</span> <span class="c1">#rRNA file to have Snoscan find snoRNAs</span>
<span class="nv">snoscan_meth</span><span class="o">=</span> <span class="c1">#-O-methylation site fileto have Snoscan find snoRNAs</span>
<span class="nv">unmask</span><span class="o">=</span><span class="m">0</span> <span class="c1">#also run ab-initio prediction programs on unmasked sequence, 1 = yes, 0 = no</span>
<span class="nv">allow_overlap</span><span class="o">=</span><span class="m">0</span> <span class="c1">#allowed gene overlap fraction (value from 0 to 1, blank for default)</span>

<span class="nv">cpus</span><span class="o">=</span><span class="m">12</span> <span class="c1">#max number of cpus to use in BLAST and RepeatMasker (not for MPI, leave 1 when using MPI)</span>

<span class="c1">#-----MAKER Behavior Options</span>
<span class="nv">max_dna_len</span><span class="o">=</span><span class="m">100000</span> <span class="c1">#length for dividing up contigs into chunks (increases/decreases memory usage)</span>
<span class="nv">min_contig</span><span class="o">=</span><span class="m">1</span> <span class="c1">#skip genome contigs below this length (under 10kb are often useless)</span>

<span class="nv">pred_flank</span><span class="o">=</span><span class="m">200</span> <span class="c1">#flank for extending evidence clusters sent to gene predictors</span>
<span class="nv">pred_stats</span><span class="o">=</span><span class="m">0</span> <span class="c1">#report AED and QI statistics for all predictions as well as models</span>
<span class="nv">AED_threshold</span><span class="o">=</span><span class="m">1</span> <span class="c1">#Maximum Annotation Edit Distance allowed (bound by 0 and 1)</span>
<span class="nv">min_protein</span><span class="o">=</span><span class="m">0</span> <span class="c1">#require at least this many amino acids in predicted proteins</span>
<span class="nv">alt_splice</span><span class="o">=</span><span class="m">0</span> <span class="c1">#Take extra steps to try and find alternative splicing, 1 = yes, 0 = no</span>
<span class="nv">always_complete</span><span class="o">=</span><span class="m">0</span> <span class="c1">#extra steps to force start and stop codons, 1 = yes, 0 = no</span>
<span class="nv">map_forward</span><span class="o">=</span><span class="m">0</span> <span class="c1">#map names and attributes forward from old GFF3 genes, 1 = yes, 0 = no</span>
<span class="nv">keep_preds</span><span class="o">=</span><span class="m">0</span> <span class="c1">#Concordance threshold to add unsupported gene prediction (bound by 0 and 1)</span>

<span class="nv">split_hit</span><span class="o">=</span><span class="m">10000</span> <span class="c1">#length for the splitting of hits (expected max intron size for evidence alignments)</span>
<span class="nv">min_intron</span><span class="o">=</span><span class="m">20</span> <span class="c1">#minimum intron length (used for alignment polishing)</span>
<span class="nv">single_exon</span><span class="o">=</span><span class="m">0</span> <span class="c1">#consider single exon EST evidence when generating annotations, 1 = yes, 0 = no</span>
<span class="nv">single_length</span><span class="o">=</span><span class="m">250</span> <span class="c1">#min length required for single exon ESTs if &#39;single_exon is enabled&#39;</span>
<span class="nv">correct_est_fusion</span><span class="o">=</span><span class="m">0</span> <span class="c1">#limits use of ESTs in annotation to avoid fusion genes</span>

<span class="nv">tries</span><span class="o">=</span><span class="m">2</span> <span class="c1">#number of times to try a contig if there is a failure for some reason</span>
<span class="nv">clean_try</span><span class="o">=</span><span class="m">0</span> <span class="c1">#remove all data from previous run before retrying, 1 = yes, 0 = no</span>
<span class="nv">clean_up</span><span class="o">=</span><span class="m">0</span> <span class="c1">#removes theVoid directory with individual analysis files, 1 = yes, 0 = no</span>
<span class="nv">TMP</span><span class="o">=</span> <span class="c1">#specify a directory other than the system default temporary directory for temporary files</span>
</code></pre></div>
<h3 id="34">3.4 迭代和整理结果<a class="headerlink" href="#34" title="Permanent link">&para;</a></h3>
<p>通过不断调整参数和迭代，可以获得多个基因组注释结果。最后一次的结果不一定是最好的，需要进行评估。</p>
<div class="highlight"><pre><span></span><code>AED_cdf_generator.pl -b <span class="m">0</span>.025 roundN.full.gff   &gt; test.roundN.AED.distrubtion.txt
</code></pre></div>
<p>可以通过AED的值来评估注释的结果。</p>
<p>AED值在0-1之间，数字越小，代表该基因注释的越准确。一般认为AED&lt;0.5即为比较好的结果。</p>
<p>重新进行BUSCO的注释，这一次用蛋白水平的注释结果文件作为输入，统计BUSCO的评分。</p>
<div class="highlight"><pre><span></span><code>busco -m prot -i test.all.protein.fasta  -l /fungi_odb10 -o busco4 -c <span class="m">12</span>  --offline <span class="p">&amp;</span>
</code></pre></div>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2016 - 2021 HZCDC@Lab
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/vendor.18f0862e.min.js"></script>
      <script src="../../assets/javascripts/bundle.994580cf.min.js"></script><script id="__lang" type="application/json">{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing"}</script>
      
      <script>
        app = initialize({
          base: "../..",
          features: ['navigation.tabs', 'navigation.sections'],
          search: Object.assign({
            worker: "../../assets/javascripts/worker/search.9c0e82ba.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script>
      
        <script src="../../javascripts/config.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>