
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="多组学联合分析">
      
      
      
        <meta name="author" content="xanadu314">
      
      
        <link rel="canonical" href="https://xanadu314.github.io/pathongs/book/C18_Analysis-Examples/02_comparative-analysis-Ban-genomes/">
      
      <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-6.2.8">
    
    
      
        <title>02. 细菌比较基因组学研究 - HYX's Bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.cb6bc1d0.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.39b8e14a.min.css">
        
          
          
          <meta name="theme-color" content="#4051b5">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto",-apple-system,BlinkMacSystemFont,Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono",SFMono-Regular,Consolas,Menlo,monospace}</style>
      
    
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/extra.css">
    
    
      
        
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-55420348-3","auto"),ga("set","anonymizeIp",!0),ga("send","pageview"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){if(this.value){var e=document.location.pathname;ga("send","pageview",e+"?q="+this.value)}})}),document.addEventListener("DOMContentSwitch",function(){ga("send","pageview",document.location.pathname)})</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
      
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="">
      
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#snps" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid" aria-label="Header">
    <a href="https://xanadu314.github.io/pathongs/" title="HYX&#39;s Bioinformatics" class="md-header-nav__button md-logo" aria-label="HYX's Bioinformatics">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header-nav__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header-nav__title" data-md-component="header-title">
      <div class="md-header-nav__ellipsis">
        <div class="md-header-nav__topic">
          <span class="md-ellipsis">
            HYX's Bioinformatics
          </span>
        </div>
        <div class="md-header-nav__topic">
          <span class="md-ellipsis">
            
              02. 细菌比较基因组学研究
            
          </span>
        </div>
      </div>
    </div>
    
      <label class="md-header-nav__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" data-md-component="search-reset" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header-nav__source">
        
<a href="https://github.com/indexofire/pathongs/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../../.." class="md-tabs__link">
      首页
    </a>
  </li>

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../main/" class="md-tabs__link md-tabs__link--active">
        数据分析笔记
      </a>
    </li>
  

      
        
  
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../../../tutorial/Linux/01_linux-installation/" class="md-tabs__link">
        专项教程
      </a>
    </li>
  

  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../protocol/" class="md-tabs__link">
        实验手册
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../blog/my-vim-ide/" class="md-tabs__link">
        生信笔记
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../about/" class="md-tabs__link">
        关于我们
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://xanadu314.github.io/pathongs/" title="HYX&#39;s Bioinformatics" class="md-nav__button md-logo" aria-label="HYX's Bioinformatics">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    HYX's Bioinformatics
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/indexofire/pathongs/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        首页
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
      
      <label class="md-nav__link" for="nav-2">
        数据分析笔记
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="数据分析笔记" data-md-level="1">
        <label class="md-nav__title" for="nav-2">
          <span class="md-nav__icon md-icon"></span>
          数据分析笔记
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../main/" class="md-nav__link">
        第零章 开始之前
      </a>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-2" type="checkbox" id="nav-2-2" >
      
      <label class="md-nav__link" for="nav-2-2">
        第一章 常用工具
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第一章 常用工具" data-md-level="2">
        <label class="md-nav__title" for="nav-2-2">
          <span class="md-nav__icon md-icon"></span>
          第一章 常用工具
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C01_Common-Utility/01_database/" class="md-nav__link">
        01. 基因组公共数据库
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C01_Common-Utility/02_get-data/" class="md-nav__link">
        02. 下载基因组数据
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C01_Common-Utility/03_entrez/" class="md-nav__link">
        03. Entrez 工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C01_Common-Utility/04_emboss/" class="md-nav__link">
        04. EMBOSS 工具集
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C01_Common-Utility/05_blast/" class="md-nav__link">
        05. Blast 比对
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C01_Common-Utility/06_fastqc/" class="md-nav__link">
        06. FastQC
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C01_Common-Utility/07_multiqc/" class="md-nav__link">
        07. Multiqc
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C01_Common-Utility/08_seqkit/" class="md-nav__link">
        08. SeqKit 工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C01_Common-Utility/09_kmer/" class="md-nav__link">
        09. Kmer 工具
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-3" type="checkbox" id="nav-2-3" >
      
      <label class="md-nav__link" for="nav-2-3">
        第二章 组装与比对
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第二章 组装与比对" data-md-level="2">
        <label class="md-nav__title" for="nav-2-3">
          <span class="md-nav__icon md-icon"></span>
          第二章 组装与比对
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C02_Raw-Data-to-Genome/01_bwa-samtools-pipeline/" class="md-nav__link">
        01. 基本比对流程
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C02_Raw-Data-to-Genome/02_quast/" class="md-nav__link">
        02. Quast 验证
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C02_Raw-Data-to-Genome/03_bacteria-assembly/" class="md-nav__link">
        03. 细菌基因组组装
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C02_Raw-Data-to-Genome/04_virus-assembly/" class="md-nav__link">
        04. 病毒基因组组装
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C02_Raw-Data-to-Genome/05_gatk3/" class="md-nav__link">
        05. GATK3
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C02_Raw-Data-to-Genome/06_gatk4/" class="md-nav__link">
        06. GATK4
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C02_Raw-Data-to-Genome/11_long-noisy-reads-assembly/" class="md-nav__link">
        11. 长序列比对
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C02_Raw-Data-to-Genome/12_hybrid-assembly/" class="md-nav__link">
        12. 二代三代混合组装
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C02_Raw-Data-to-Genome/14_bwa-long-sequence-mapping/" class="md-nav__link">
        14. bwa 长序列比对
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-4" type="checkbox" id="nav-2-4" >
      
      <label class="md-nav__link" for="nav-2-4">
        第三章 基因组注释
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第三章 基因组注释" data-md-level="2">
        <label class="md-nav__title" for="nav-2-4">
          <span class="md-nav__icon md-icon"></span>
          第三章 基因组注释
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C03_Genome-Annotation/01_prokka/" class="md-nav__link">
        01. Prokka 快速注释
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C03_Genome-Annotation/02_rast/" class="md-nav__link">
        02. 在线注释工具 rast
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C03_Genome-Annotation/03_pgaap/" class="md-nav__link">
        03. 注释工具 PGAAP
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C03_Genome-Annotation/04_snpeff/" class="md-nav__link">
        04. SNP注释工具 snpEFF
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C03_Genome-Annotation/05_cog/" class="md-nav__link">
        05. COG注释
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-5" type="checkbox" id="nav-2-5" >
      
      <label class="md-nav__link" for="nav-2-5">
        第四章 基因组比较
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第四章 基因组比较" data-md-level="2">
        <label class="md-nav__title" for="nav-2-5">
          <span class="md-nav__icon md-icon"></span>
          第四章 基因组比较
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C06_Genome-Compare/01_mummer/" class="md-nav__link">
        01. mummer
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C06_Genome-Compare/02_nucdiff/" class="md-nav__link">
        02. nucdiff
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C06_Genome-Compare/03_mauve/" class="md-nav__link">
        03. mauve
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C06_Genome-Compare/04_sibelia/" class="md-nav__link">
        04. sibelia
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C06_Genome-Compare/05_ani/" class="md-nav__link">
        05. ani
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C06_Genome-Compare/10_bigsi/" class="md-nav__link">
        10. bigsi
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-6" type="checkbox" id="nav-2-6" >
      
      <label class="md-nav__link" for="nav-2-6">
        第五章 泛基因组分析
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第五章 泛基因组分析" data-md-level="2">
        <label class="md-nav__title" for="nav-2-6">
          <span class="md-nav__icon md-icon"></span>
          第五章 泛基因组分析
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C07_PanGenome-and-CoreGenome/01_pangenome/" class="md-nav__link">
        01. Pangenome 概述
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C07_PanGenome-and-CoreGenome/02_get_homologues/" class="md-nav__link">
        02. Get_Homologues 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C07_PanGenome-and-CoreGenome/03_bpga/" class="md-nav__link">
        03. BPGA 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C07_PanGenome-and-CoreGenome/04_roary/" class="md-nav__link">
        04. Roary 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C07_PanGenome-and-CoreGenome/05_panseq/" class="md-nav__link">
        05. panseq 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C07_PanGenome-and-CoreGenome/07_orthmcl/" class="md-nav__link">
        07. OrthMCL 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C07_PanGenome-and-CoreGenome/08_itep/" class="md-nav__link">
        08. ITEP 分析
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-7" type="checkbox" id="nav-2-7" >
      
      <label class="md-nav__link" for="nav-2-7">
        第六章 基因组可视化
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第六章 基因组可视化" data-md-level="2">
        <label class="md-nav__title" for="nav-2-7">
          <span class="md-nav__icon md-icon"></span>
          第六章 基因组可视化
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C08_Genome-Visualization/01_igv/" class="md-nav__link">
        01. igv 查看比对结果
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C08_Genome-Visualization/02_brig/" class="md-nav__link">
        02. BRIG 分析工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C08_Genome-Visualization/03_circos/" class="md-nav__link">
        03. circos 工具使用
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-8" type="checkbox" id="nav-2-8" >
      
      <label class="md-nav__link" for="nav-2-8">
        第七章 综合分析平台
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第七章 综合分析平台" data-md-level="2">
        <label class="md-nav__title" for="nav-2-8">
          <span class="md-nav__icon md-icon"></span>
          第七章 综合分析平台
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C09_Analysis-Platform/01_docker/" class="md-nav__link">
        01. 基于 docker 镜像
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C09_Analysis-Platform/02_galaxy/" class="md-nav__link">
        02. Galaxy 平台
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C09_Analysis-Platform/03_amazon/" class="md-nav__link">
        03. Amazon EC2 云平台
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C09_Analysis-Platform/04_others/" class="md-nav__link">
        04. 其他平台
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C16_Standalone-Platform/01_anvio/" class="md-nav__link">
        05. anvio
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C16_Standalone-Platform/02_galaxy/" class="md-nav__link">
        02. galaxy
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-9" type="checkbox" id="nav-2-9" >
      
      <label class="md-nav__link" for="nav-2-9">
        第八章 重组与种群分析
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第八章 重组与种群分析" data-md-level="2">
        <label class="md-nav__title" for="nav-2-9">
          <span class="md-nav__icon md-icon"></span>
          第八章 重组与种群分析
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C10_Recombination-and-Population-Structure/01_virus-recombination/" class="md-nav__link">
        01. 病毒重组
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C10_Recombination-and-Population-Structure/02_gubbins/" class="md-nav__link">
        02. Gubbins 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C10_Recombination-and-Population-Structure/03_baps/" class="md-nav__link">
        03. BAPS
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-10" type="checkbox" id="nav-2-10" >
      
      <label class="md-nav__link" for="nav-2-10">
        第九章 细菌扫描工具
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第九章 细菌扫描工具" data-md-level="2">
        <label class="md-nav__title" for="nav-2-10">
          <span class="md-nav__icon md-icon"></span>
          第九章 细菌扫描工具
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C11_Insilico-Scanning/01_srst2/" class="md-nav__link">
        01. SRST2 短序列扫描工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C11_Insilico-Scanning/02_mlst/" class="md-nav__link">
        02. MLST 扫描工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C11_Insilico-Scanning/03_ariba/" class="md-nav__link">
        03. Ariba 耐药基因
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C11_Insilico-Scanning/04_abricate/" class="md-nav__link">
        04. Abricate 多功能扫描工具
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-11" type="checkbox" id="nav-2-11" >
      
      <label class="md-nav__link" for="nav-2-11">
        第十章 宏基因组分析
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十章 宏基因组分析" data-md-level="2">
        <label class="md-nav__title" for="nav-2-11">
          <span class="md-nav__icon md-icon"></span>
          第十章 宏基因组分析
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C12_Metagenomics-Analysis/01_kraken/" class="md-nav__link">
        01. kraken
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C12_Metagenomics-Analysis/02_centrifuge/" class="md-nav__link">
        02. centrifuge
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C12_Metagenomics-Analysis/03_bracken/" class="md-nav__link">
        03. bracken
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C12_Metagenomics-Analysis/04_sourmash/" class="md-nav__link">
        04. sourmash
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C12_Metagenomics-Analysis/05_clark/" class="md-nav__link">
        05. clark
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C12_Metagenomics-Analysis/06_krona/" class="md-nav__link">
        06. krona
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C12_Metagenomics-Analysis/07_pavian/" class="md-nav__link">
        07. pavian
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C12_Metagenomics-Analysis/08_supri/" class="md-nav__link">
        08. supri
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C12_Metagenomics-Analysis/09_shotgun/" class="md-nav__link">
        09. shotgun 分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C12_Metagenomics-Analysis/10_16s/" class="md-nav__link">
        10. 16S 种群分析
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C12_Metagenomics-Analysis/11_kmerfinder/" class="md-nav__link">
        11. KmerFinder
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C12_Metagenomics-Analysis/12_meta-beetl/" class="md-nav__link">
        12. Meta-BEETL
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-12" type="checkbox" id="nav-2-12" >
      
      <label class="md-nav__link" for="nav-2-12">
        第十一章 单分子测序
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十一章 单分子测序" data-md-level="2">
        <label class="md-nav__title" for="nav-2-12">
          <span class="md-nav__icon md-icon"></span>
          第十一章 单分子测序
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C13_Single-Molecular-Sequencing/01_nanopore/" class="md-nav__link">
        01. nanopore
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C13_Single-Molecular-Sequencing/02_pacbio/" class="md-nav__link">
        02. pacbio
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C13_Single-Molecular-Sequencing/03_canu/" class="md-nav__link">
        03. canu
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C13_Single-Molecular-Sequencing/04_seqs-polish/" class="md-nav__link">
        04. trimming
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C13_Single-Molecular-Sequencing/06_hybrid-assembling/" class="md-nav__link">
        06. hybrid
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-13" type="checkbox" id="nav-2-13" >
      
      <label class="md-nav__link" for="nav-2-13">
        第十二章 多重序列比对
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十二章 多重序列比对" data-md-level="2">
        <label class="md-nav__title" for="nav-2-13">
          <span class="md-nav__icon md-icon"></span>
          第十二章 多重序列比对
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C14_Multiple-Sequence-Alignment/01_msa/" class="md-nav__link">
        01. 多重序列比对
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-14" type="checkbox" id="nav-2-14" >
      
      <label class="md-nav__link" for="nav-2-14">
        第十三章 进化树构建分析
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十三章 进化树构建分析" data-md-level="2">
        <label class="md-nav__title" for="nav-2-14">
          <span class="md-nav__icon md-icon"></span>
          第十三章 进化树构建分析
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C15_Phylogeny-Analysis/00_all-about-phylogeny/" class="md-nav__link">
        00. 构建进化书
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C15_Phylogeny-Analysis/02_raxml.md" class="md-nav__link">
        01. RAxML
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C15_Phylogeny-Analysis/01_figtree.md" class="md-nav__link">
        02. figtree
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C15_Phylogeny-Analysis/02_ggtree.md" class="md-nav__link">
        03. ggtree
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-15" type="checkbox" id="nav-2-15" >
      
      <label class="md-nav__link" for="nav-2-15">
        第十七章 分子地理流行病学
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十七章 分子地理流行病学" data-md-level="2">
        <label class="md-nav__title" for="nav-2-15">
          <span class="md-nav__icon md-icon"></span>
          第十七章 分子地理流行病学
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C17_Pathogen-Geospatial-Analysis/04_cartodb/" class="md-nav__link">
        04. CartoDB
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C17_Pathogen-Geospatial-Analysis/05_leaflet/" class="md-nav__link">
        05. leaflet
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-16" type="checkbox" id="nav-2-16" checked>
      
      <label class="md-nav__link" for="nav-2-16">
        第十八章 分析实践
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十八章 分析实践" data-md-level="2">
        <label class="md-nav__title" for="nav-2-16">
          <span class="md-nav__icon md-icon"></span>
          第十八章 分析实践
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../01_trace-Pae-from-different-source-by-SNP/" class="md-nav__link">
        01. 细菌SNP溯源分析一
      </a>
    </li>
  

          
            
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          02. 细菌比较基因组学研究
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        02. 细菌比较基因组学研究
      </a>
      
        
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1" class="md-nav__link">
    1. 获取公共数据库数据
  </a>
  
    <nav class="md-nav" aria-label="1. 获取公共数据库数据">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-sra" class="md-nav__link">
    1.1 下载 sra
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#12-pese" class="md-nav__link">
    1.2 区分 PE/SE 数据
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#13-fastqgz" class="md-nav__link">
    1.3 验证 fastq.gz 文件名的一致性
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" class="md-nav__link">
    2. 数据的前期处理
  </a>
  
    <nav class="md-nav" aria-label="2. 数据的前期处理">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-qc" class="md-nav__link">
    2.1 数据 QC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22" class="md-nav__link">
    2.2 去除接头
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-pangenome" class="md-nav__link">
    3. Pangenome 分析
  </a>
  
    <nav class="md-nav" aria-label="3. Pangenome 分析">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31" class="md-nav__link">
    3.1 基因组组装
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32" class="md-nav__link">
    3.2 基因组注释
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#33-roary" class="md-nav__link">
    3.3 用 Roary 进行分析
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#34-harvest" class="md-nav__link">
    3.4 Harvest
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-snps" class="md-nav__link">
    4. SNPs 构建进化树
  </a>
  
    <nav class="md-nav" aria-label="4. SNPs 构建进化树">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41-mapping" class="md-nav__link">
    4.1 基于参考基因组 Mapping 的方法
  </a>
  
    <nav class="md-nav" aria-label="4.1 基于参考基因组 Mapping 的方法">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-snippy" class="md-nav__link">
    1. Snippy*
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5" class="md-nav__link">
    5. 核心基因分析
  </a>
  
    <nav class="md-nav" aria-label="5. 核心基因分析">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#51-get_homologues" class="md-nav__link">
    5.1 get_homologues
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6" class="md-nav__link">
    6. 相关软件安装
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-reference" class="md-nav__link">
    7. Reference
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-17" type="checkbox" id="nav-2-17" >
      
      <label class="md-nav__link" for="nav-2-17">
        第十九章 工具流工具
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第十九章 工具流工具" data-md-level="2">
        <label class="md-nav__title" for="nav-2-17">
          <span class="md-nav__icon md-icon"></span>
          第十九章 工具流工具
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C19_Pipelines-and-Workflows/05_workflow-software/" class="md-nav__link">
        05. workflow工具软件
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-18" type="checkbox" id="nav-2-18" >
      
      <label class="md-nav__link" for="nav-2-18">
        第二十章 其他
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="第二十章 其他" data-md-level="2">
        <label class="md-nav__title" for="nav-2-18">
          <span class="md-nav__icon md-icon"></span>
          第二十章 其他
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../C20_Others/02_gi-to-accession/" class="md-nav__link">
        02. GI转Accession
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" >
      
      <label class="md-nav__link" for="nav-3">
        专项教程
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="专项教程" data-md-level="1">
        <label class="md-nav__title" for="nav-3">
          <span class="md-nav__icon md-icon"></span>
          专项教程
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-3-1" type="checkbox" id="nav-3-1" >
      
      <label class="md-nav__link" for="nav-3-1">
        Linux入门
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Linux入门" data-md-level="2">
        <label class="md-nav__title" for="nav-3-1">
          <span class="md-nav__icon md-icon"></span>
          Linux入门
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Linux/01_linux-installation/" class="md-nav__link">
        01. Linux 安装示例
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Linux/02_basic-cml/" class="md-nav__link">
        02. Linux 基本命令
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Linux/03_linux-software/" class="md-nav__link">
        03. Linux 软件管理
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Linux/04_awk/" class="md-nav__link">
        04. awk 列处理工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Linux/05_sed/" class="md-nav__link">
        05. sed 行处理工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Linux/06_tools/" class="md-nav__link">
        06. 其他命令行工具
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Linux/07_vim/" class="md-nav__link">
        07. vim
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-3-2" type="checkbox" id="nav-3-2" >
      
      <label class="md-nav__link" for="nav-3-2">
        常用编程语言
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="常用编程语言" data-md-level="2">
        <label class="md-nav__title" for="nav-3-2">
          <span class="md-nav__icon md-icon"></span>
          常用编程语言
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Programming/01_shell/" class="md-nav__link">
        01. Shell 脚本
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Programming/02_perl/" class="md-nav__link">
        02. Perl 编程
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Programming/03_python/" class="md-nav__link">
        03. Python 编程
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Programming/04_biopython/" class="md-nav__link">
        04. Biopython 编程
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Programming/05_r/" class="md-nav__link">
        05. R 编程
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-3-3" type="checkbox" id="nav-3-3" >
      
      <label class="md-nav__link" for="nav-3-3">
        数据文件格式
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="数据文件格式" data-md-level="2">
        <label class="md-nav__title" for="nav-3-3">
          <span class="md-nav__icon md-icon"></span>
          数据文件格式
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Format/01_fasta/" class="md-nav__link">
        01. fasta
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Format/02_embl/" class="md-nav__link">
        02. embl
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Format/05_fastq.md" class="md-nav__link">
        03. fastq
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Format/04_gcg/" class="md-nav__link">
        04. gcg
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Format/05_sam/" class="md-nav__link">
        05. sam
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Format/06_bam/" class="md-nav__link">
        06. bam
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Format/07_bed/" class="md-nav__link">
        07. bed
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Format/08_vcf/" class="md-nav__link">
        08. vcf
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Format/09_gff/" class="md-nav__link">
        09. gff
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../tutorial/Format/10_cram/" class="md-nav__link">
        10. cram
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4" >
      
      <label class="md-nav__link" for="nav-4">
        实验手册
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="实验手册" data-md-level="1">
        <label class="md-nav__title" for="nav-4">
          <span class="md-nav__icon md-icon"></span>
          实验手册
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../protocol/" class="md-nav__link">
        
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../protocol/artic-network-covid19/" class="md-nav__link">
        ArticNetwork Covid-19测序
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" >
      
      <label class="md-nav__link" for="nav-5">
        生信笔记
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="生信笔记" data-md-level="1">
        <label class="md-nav__title" for="nav-5">
          <span class="md-nav__icon md-icon"></span>
          生信笔记
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../blog/my-vim-ide/" class="md-nav__link">
        01 TEST
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../blog/blast2sra/" class="md-nav__link">
        02 blast2SRA
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../blog/tes/" class="md-nav__link">
        03 Rmarkdown
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6" >
      
      <label class="md-nav__link" for="nav-6">
        关于我们
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="关于我们" data-md-level="1">
        <label class="md-nav__title" for="nav-6">
          <span class="md-nav__icon md-icon"></span>
          关于我们
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../about/" class="md-nav__link">
        
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1" class="md-nav__link">
    1. 获取公共数据库数据
  </a>
  
    <nav class="md-nav" aria-label="1. 获取公共数据库数据">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-sra" class="md-nav__link">
    1.1 下载 sra
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#12-pese" class="md-nav__link">
    1.2 区分 PE/SE 数据
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#13-fastqgz" class="md-nav__link">
    1.3 验证 fastq.gz 文件名的一致性
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" class="md-nav__link">
    2. 数据的前期处理
  </a>
  
    <nav class="md-nav" aria-label="2. 数据的前期处理">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-qc" class="md-nav__link">
    2.1 数据 QC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22" class="md-nav__link">
    2.2 去除接头
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-pangenome" class="md-nav__link">
    3. Pangenome 分析
  </a>
  
    <nav class="md-nav" aria-label="3. Pangenome 分析">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31" class="md-nav__link">
    3.1 基因组组装
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32" class="md-nav__link">
    3.2 基因组注释
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#33-roary" class="md-nav__link">
    3.3 用 Roary 进行分析
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#34-harvest" class="md-nav__link">
    3.4 Harvest
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-snps" class="md-nav__link">
    4. SNPs 构建进化树
  </a>
  
    <nav class="md-nav" aria-label="4. SNPs 构建进化树">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41-mapping" class="md-nav__link">
    4.1 基于参考基因组 Mapping 的方法
  </a>
  
    <nav class="md-nav" aria-label="4.1 基于参考基因组 Mapping 的方法">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-snippy" class="md-nav__link">
    1. Snippy*
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5" class="md-nav__link">
    5. 核心基因分析
  </a>
  
    <nav class="md-nav" aria-label="5. 核心基因分析">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#51-get_homologues" class="md-nav__link">
    5.1 get_homologues
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6" class="md-nav__link">
    6. 相关软件安装
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-reference" class="md-nav__link">
    7. Reference
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/indexofire/pathongs/edit/master/docs/book/C18_Analysis-Examples/02_comparative-analysis-Ban-genomes.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                <h1 id="snps">炭疽芽胞杆菌基因组 SNPs 分析<a class="headerlink" href="#snps" title="Permanent link">&para;</a></h1>
<p>{{ git_page_authors }} 更新于: {{ git_revision_date }}</p>
<hr />
<div class="admonition abstract">
<p class="admonition-title">内容简介</p>
<p>本节主要通过构建相应的分析流程，获得炭疽芽胞杆菌菌株基因组数据，可以第一时间就进行溯源方面的研究。将这个工作的过程建立文档保存。分析流程粗略简单，主要目的也是为实验室内部教学，希望能做到举一反三，给其他细菌性病原微生物基因组溯源分析研究提供帮助。</p>
</div>
<p>做为疾控中心的实验室，对生物恐怖事件的防控和应对处置能力是应急工作能力的重要组成之一。而细菌性生物恐怖战剂中最著名的就是含有炭疽芽胞杆菌的白色粉末。做为基层实验室，因为在日常工作中接触到这些样品和病原的机会微乎其微，所以经验和技能方面较为缺乏。通过平时加强相关项目的应急演练外，实验室快速分析和溯源工作需要做好能力储备，因此实验室对炭疽芽胞杆菌的基因组数据进行了基本分析，对相关资料和信息进行了储备。</p>
<p>软件中使用的一些参数是根据自己的服务器设置的，具体运行时需要根据计算机配置进行相应的调整。</p>
<div class="admonition note">
<p class="admonition-title">服务器配置</p>
<ul>
<li>CPU: E3 2650v3 x2</li>
<li>Memory: ECC 16G x8</li>
<li>Harddisk: 7200转 1T x4 组 Raid 5 + 1 Hotspare</li>
<li>System: Ubuntu 16.04 LTS 64bit</li>
</ul>
</div>
<h2 id="1">1. 获取公共数据库数据<a class="headerlink" href="#1" title="Permanent link">&para;</a></h2>
<p>数据采用 NCBI SRA 数据库中的公共数据来作分析。首先检索 NCBI SRA 数据库中所有使用 illumina 平台对炭疽芽胞杆菌基因组进行 PE 测序的 大于 100M 的 NGS 数据，并在下载这些 sra 数据后转换成 fastq.gz 格式。使用的工具主要为 <code>edirect</code> 和 <code>sra-tools</code>。</p>
<h3 id="11-sra">1.1 下载 sra<a class="headerlink" href="#11-sra" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="c1"># 下载炭疽芽胞杆菌 SRA 双末端测序数据</span>
$ esearch -db sra -query <span class="s1">&#39;&quot;Bacillus anthracis&quot;[Organism] AND \</span>
<span class="s1">&gt; (Hiseq[ALL] OR Miseq[ALL]) AND &quot;strategy wgs&quot;[Properties] AND paired[ALL]&#39;</span> <span class="p">|</span> <span class="se">\</span>
&gt; efetch -db sra -format runinfo <span class="p">|</span> grep <span class="s1">&#39;^[DSE]RR*&#39;</span> <span class="p">|</span> <span class="se">\</span>
&gt; awk -F<span class="s1">&#39;,&#39;</span> <span class="s1">&#39;{if($5&gt;100000000) print $1}&#39;</span> <span class="p">|</span> sort -n <span class="p">|</span> <span class="se">\</span>
&gt; xargs fastq-dump --split-files --gzip --outdir bacillus_anthracis
</code></pre></div>
<p>上面的命令会直接调用 <code>fastq-dump</code> 下载数据到 <code>--outdir</code> 中。虽然我们用了 <code>grep PAIRED</code> 来区分不同的测序方式，但是由于信息是提交者上传的，可能信息不一定是正确的。由于后面的流程中牵涉到组装和比对等操作，PE/SE的方式会有区别，最好还是进行验证后再下载，或者将PE/SE的数据区分存放。</p>
<div class="highlight"><pre><span></span><code><span class="c1"># 先生成数据记录</span>
$ esearch -db sra -query <span class="s1">&#39;&quot;Bacillus anthracis&quot;[Organism] AND \</span>
<span class="s1">&gt; (Hiseq[ALL] OR Miseq[ALL]) AND &quot;strategy wgs&quot;[Properties]&#39;</span> <span class="p">|</span> <span class="se">\</span>
&gt; efetch -db sra -format runinfo <span class="p">|</span> grep <span class="s1">&#39;PAIRED&#39;</span> <span class="p">|</span> grep <span class="s1">&#39;^[DSE]RR*&#39;</span> <span class="p">|</span> <span class="se">\</span>
&gt; awk -F<span class="s1">&#39;,&#39;</span> <span class="s1">&#39;{if($5&gt;100000000) print $1}&#39;</span> <span class="p">|</span> sort -n &gt; ban-sra.txt

<span class="c1"># 下载 sra</span>
$ cat ban-sra.txt <span class="p">|</span> xargs prefetch
</code></pre></div>
<p>这样会下载 sra 文件到默认目录(~/.ncbi/public/sra)，然后用下面的方法来区分下载的数据到底是PE还是SE测序。</p>
<h3 id="12-pese">1.2 区分 PE/SE 数据<a class="headerlink" href="#12-pese" title="Permanent link">&para;</a></h3>
<p>虽然我们检索时限定了 PE 测序数据，但是很多时候数据库里的信息不准确，还是会有很多 SE 的数据会被下载。确定 PE 或者 SE 的方法很多。</p>
<div class="highlight"><pre><span></span><code><span class="c1"># sra-stat 统计 sra 文件，nreads=1 表示 SE 测序，nreads=2 表示 PE 测序。-e 2 只统计前2个spot，如果去掉-e参数，程序统计整个文件，速度比较慢</span>
$ <span class="k">for</span> i <span class="k">in</span> *.sra<span class="p">;</span> <span class="k">do</span> <span class="nb">echo</span> <span class="nv">$i</span> <span class="sb">`</span>sra-stat -xs -e <span class="m">2</span> <span class="nv">$i</span> <span class="p">|</span> grep <span class="s2">&quot;nreads&quot;</span><span class="sb">`</span><span class="p">;</span> <span class="k">done</span>

<span class="c1"># 批量删除</span>
$ parallel <span class="s1">&#39;sra-stat -xs -e 2&#39;</span> ::: *.sra <span class="p">|</span> xtract -Pattern Run -element Run@accession <span class="se">\</span>
&gt; Statistics@nreads <span class="p">|</span> awk <span class="s1">&#39;{if($2==1) print $1}&#39;</span> <span class="p">|</span> xargs rm
</code></pre></div>
<p><code>sra-stat</code> 工具可以提取 sra 文件的测序相关信息。<code>-xs</code> 参数生成 xml 格式的统计信息。里面属性值 nreads 可以表示采用的是哪一种测序方式。如果 spot-id 是规则排列，那么可以添加 <code>-e 2</code> 最快时间获得结果。</p>
<p><code>parallel</code> 工具来并行化处理 <code>sra-stat</code>，如果计算量大是很好的利用多核CPU处理终端命令的方法。</p>
<div class="highlight"><pre><span></span><code><span class="c1"># fastq-dump 输出一个 reads，计算行数如果是4，那么就是 SE，如果是8,则是 PE</span>
$ fastq-dump -X <span class="m">1</span> --split-spot -Z SRR955386.sra <span class="p">|</span> wc -l

<span class="c1"># 利用 -A 参数远程获得结果</span>
$ fastq-dump -X1 --split-spot -Z -L <span class="m">5</span> -A SRR955386

<span class="c1"># fastq-dump 自己判断，当 sra 文件是 SE 测序时，fastq-dump 只能生成1个 *_1.fastq 文件</span>
$ fastq-dump --split-files ERR493452.sra

<span class="c1"># fastq-dump 只能调用单核转换，可以使用 parallel 工具并行处理加快速度。</span>
$ parallel <span class="s2">&quot;fastq-dump --split-files --gzip --outdir bacillus_anthracis&quot;</span> ::: *.sra
</code></pre></div>
<p>另外一种获得是 PE/SE 的方法是用 <code>fastq-dump</code> 来读取出相同 spot-id 的 reads 再计算行数。<code>-X</code> 表示最多读取的 spot-id 数量；<code>--split-spot</code> 区分spot到reads然后用<code>-Z</code>输出，再<code>wc -l</code>统计行数。</p>
<p>有时候有些sra文件在 dump 时会出错，不能生成对应的 PE fastq 文件，这就需要比较后删除了。如果是 mate pair 的数据，会生成 <code>*_3.fastq.gz</code>，如果不需要则可手工删除。</p>
<h3 id="13-fastqgz">1.3 验证 fastq.gz 文件名的一致性<a class="headerlink" href="#13-fastqgz" title="Permanent link">&para;</a></h3>
<p><code>fastq-dump</code> 批量生成 fastq 数据后，有时因为各种原因，还是会 dump 出非 PE 类型的 <code>*._1</code> 和 <code>*._2</code> 命名风格的2个PE数据文件。文件量大时通过肉眼很难判断。</p>
<div class="highlight"><pre><span></span><code><span class="c1"># 核对 PE 双端测序文件是否一致</span>
$ ls -l *_1.fastq.gz <span class="p">|</span> awk <span class="s1">&#39;{print $9}&#39;</span> <span class="p">|</span> awk -F<span class="s1">&#39;_&#39;</span> <span class="s1">&#39;{print $1}&#39;</span> &gt; R1.txt
$ ls -l *_2.fastq.gz <span class="p">|</span> awk <span class="s1">&#39;{print $9}&#39;</span> <span class="p">|</span> awk -F<span class="s1">&#39;_&#39;</span> <span class="s1">&#39;{print $1}&#39;</span> &gt; R2.txt
$ diff R1.txt R2.txt
</code></pre></div>
<p>通过 <code>diff</code> 工具比较，如果2个文件内容一致，则终端不会有输出。否则就要删除差异的那个文件了。</p>
<h2 id="2">2. 数据的前期处理<a class="headerlink" href="#2" title="Permanent link">&para;</a></h2>
<p>由于是公共数据库下载的数据，并不能保证测序实验质量或者数据提交者是否提交的是质控后的数据。因此建议要对数据做一些前期的质控处理。</p>
<h3 id="21-qc">2.1 数据 QC<a class="headerlink" href="#21-qc" title="Permanent link">&para;</a></h3>
<p>首先对基因组 GC 含量和 Q 值进行初步筛选，对于偏差较大的数据考虑直接剔除（或者用其他软件验证看是否是错误物种）。这里使用的工具为 <code>bioawk</code> 或 <code>parallel</code>。</p>
<div class="highlight"><pre><span></span><code><span class="c1"># 因为 reads 覆盖度可能并不均一，高覆盖度的区域 GC 含量比重会略高。但是大部分情况下，正常测序的基因组这种区域相对来说不多，平均到基因组后会整体 GC 含量影响不大。</span>
$ <span class="k">for</span> i <span class="k">in</span> *.fastq.gz<span class="p">;</span> <span class="k">do</span> bioawk -c fastx <span class="s1">&#39;BEGIN{n=0;q=0}{n+=gc($seq);q+=meanqual($seq)}END{print $name,n/NR,q/NR}&#39;</span> <span class="nv">$i</span> &gt;&gt; result.txt<span class="p">;</span> <span class="k">done</span>

<span class="c1"># awk 类工具是单进程的，为了加速可以使用 parallel 来并行计算</span>
$ parallel <span class="s2">&quot;bioawk -c fastx &#39;BEGIN{n=0;q=0}{n+=gc(\$seq);q+=meanqual(\$seq)}END{print \$name,n/NR,q/NR}&#39; &gt;&gt; result.txt&quot;</span> ::: *.fastq.gz

<span class="c1"># 按照 GC 含量排序</span>
$ cat result.txt <span class="p">|</span> sort -n -k3
</code></pre></div>
<p>R 绘制 gc 分布图，如果 gc 含量偏差超过 10% 时就剔除该基因组数据。</p>
<div class="highlight"><pre><span></span><code><span class="o">&gt;</span> <span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="o">&gt;</span> <span class="n">data</span><span class="o">&lt;-</span><span class="nf">read.table</span><span class="p">(</span><span class="s">&quot;result.txt&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> <span class="nf">qplot</span><span class="p">(</span><span class="n">v2</span><span class="p">,</span> <span class="n">v3</span><span class="p">,</span> <span class="n">data</span><span class="o">=</span><span class="n">data</span><span class="p">)</span>
</code></pre></div>
<p>结果显示有一个黑点的GC含量特别低，且Q值特别高。结果显示测序实验 SRR2155551 和 SRR2164197 的 GC 含量仅为22%，而其他样本 GC 含量范围在34%～41%。且这2个实验的样本均为同一个，是1个样本的2此测序（或者数据上传了2次），因此将其剔除。</p>
<h3 id="22">2.2 去除接头<a class="headerlink" href="#22" title="Permanent link">&para;</a></h3>
<p>其次要去除接头的污染。因为分析流程中不仅包括 mapping 的方式，还包含 de novo assembly，为了避免接头序列对基因组拼接的影响，这里最好进行。这里使用的工具是 <code>FastQC</code> 和 <code>fadapa</code></p>
<div class="highlight"><pre><span></span><code><span class="c1"># 使用脚本 scan_adaptors.py 来扫描下载的高通量基因组测序数据是否有接头污染的情况。</span>
$ fastqc -t <span class="m">40</span> -q --extract *.fastq.gz
$ python scan_fastqc_report.py -d qc
</code></pre></div>
<p>接头去除软件很多，但是要自动判断并且批量处理最好自己写脚本比较方便。这里用 python 写一个批量移除的脚本。</p>
<div class="highlight"><pre><span></span><code><span class="ch">#!usr/bin/env python</span>
<span class="c1"># -*- coding: utf-8 -*-</span>
<span class="c1"># scripts name: scan_fastqc_report.py</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>
<span class="kn">from</span> <span class="nn">Bio.SeqRecord</span> <span class="kn">import</span> <span class="n">SeqRecord</span>
<span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="kn">import</span> <span class="n">generic_nucleotide</span>
<span class="kn">from</span> <span class="nn">fadapa</span> <span class="kn">import</span> <span class="n">Fadapa</span>
<span class="kn">import</span> <span class="nn">argparse</span>

<span class="n">parser</span> <span class="o">=</span> <span class="n">argparse</span><span class="o">.</span><span class="n">ArgumentParser</span><span class="p">()</span>
<span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-i&#39;</span><span class="p">,</span> <span class="s1">&#39;--input&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;qcfile&#39;</span><span class="p">,</span> <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">,</span> <span class="n">help</span><span class="o">=</span><span class="s2">&quot;fastqc extract data file&quot;</span><span class="p">)</span>
<span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-o&#39;</span><span class="p">,</span> <span class="s1">&#39;--output&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;adaptor&#39;</span><span class="p">,</span> <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">,</span> <span class="n">help</span><span class="o">=</span><span class="s2">&quot;adaptors file&quot;</span><span class="p">)</span>
<span class="n">args</span> <span class="o">=</span> <span class="n">parser</span><span class="o">.</span><span class="n">parse_args</span><span class="p">()</span>
<span class="n">qc_file</span> <span class="o">=</span> <span class="n">Fadapa</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">qcfile</span><span class="p">)</span>

<span class="k">if</span> <span class="n">qc_file</span><span class="o">.</span><span class="n">summary</span><span class="p">()[</span><span class="o">-</span><span class="mi">3</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;fail&#39;</span><span class="p">:</span>
    <span class="n">adaptors</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">ors</span> <span class="o">=</span> <span class="n">qc_file</span><span class="o">.</span><span class="n">clean_data</span><span class="p">(</span><span class="s2">&quot;Overrepresented sequences&quot;</span><span class="p">)</span>
    <span class="n">ors</span><span class="o">.</span><span class="n">pop</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span>

    <span class="k">for</span> <span class="p">(</span><span class="n">index</span><span class="p">,</span> <span class="n">seq</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">ors</span><span class="p">):</span>
        <span class="k">if</span> <span class="s1">&#39;NNNNNN&#39;</span> <span class="ow">in</span> <span class="n">seq</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span>
            <span class="k">continue</span>
        <span class="n">adaptors</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">SeqRecord</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="n">seq</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">generic_nucleotide</span><span class="p">),</span> <span class="nb">id</span><span class="o">=</span><span class="s2">&quot;adaptor_</span><span class="si">%d</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">index</span><span class="o">+</span><span class="mi">1</span><span class="p">)))</span>

    <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">adaptor</span><span class="p">):</span>
        <span class="n">os</span><span class="o">.</span><span class="n">makedirs</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">adaptor</span><span class="p">)</span>
    <span class="n">SeqIO</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">adaptors</span><span class="p">,</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">/adaptor.fasta&quot;</span> <span class="o">%</span> <span class="n">args</span><span class="o">.</span><span class="n">adaptor</span><span class="p">,</span> <span class="s2">&quot;fasta&quot;</span><span class="p">)</span>
    <span class="nb">print</span> <span class="s2">&quot;Overrepresented sequences has been save to adaptors.fasta&quot;</span>
<span class="k">else</span><span class="p">:</span>
    <span class="nb">print</span> <span class="s2">&quot;No Overrepresented sequences&quot;</span>
</code></pre></div>
<h2 id="3-pangenome">3. Pangenome 分析<a class="headerlink" href="#3-pangenome" title="Permanent link">&para;</a></h2>
<p>常用的 Pangenome 分析包括基因组组装注释后对基因</p>
<h3 id="31">3.1 基因组组装<a class="headerlink" href="#31" title="Permanent link">&para;</a></h3>
<p>组装细菌基因组最常用的工具是 <code>spades</code>，它可以分批对多个kmer进行组装。</p>
<div class="highlight"><pre><span></span><code><span class="c1"># spades 组装</span>
$ <span class="k">for</span> i <span class="k">in</span> <span class="k">$(</span>awk -F<span class="s1">&#39;_&#39;</span> <span class="s1">&#39;!a[$1]++{print}&#39;</span><span class="k">)</span><span class="p">;</span> <span class="k">do</span> spades.py --careful <span class="se">\</span>
&gt; -k <span class="m">21</span>,33,55,77,99,121 -1 <span class="nv">$i_1</span>.fastq.gz -2 <span class="nv">$i_2</span>.fastq.gz -T <span class="m">40</span> -o <span class="nv">$i</span><span class="p">;</span> <span class="k">done</span>
</code></pre></div>
<p>但是由于下载的批量数据采用不同平台，在不同时间进行的测序，读长彼此差异还是比较大的。比如10K病原基因组计划里，很多样本适用 miseq 进行测序的，读长可达250以上，而很多商业公司或者科研院校则用 hiseq 测序，读长一般在100～150。用同一个参数来组装可能不会获得较好的结果。<code>shovill</code>工具可以帮我们自动化最佳kmer选择，去除低覆盖contigs等操作，一步达到较好的组装序列。</p>
<div class="highlight"><pre><span></span><code><span class="c1"># shovill 组装</span>
$ <span class="k">for</span> i <span class="k">in</span> <span class="k">$(</span>awk -F<span class="s1">&#39;_&#39;</span> <span class="s1">&#39;!a[$1]++{print}&#39;</span><span class="k">)</span><span class="p">;</span> <span class="k">do</span> shovill -R1 <span class="nv">$i_1</span>.fastq.gz -R2 <span class="nv">$i_2</span>.fastq.gz -o <span class="nv">$i</span><span class="p">;</span> <span class="k">done</span>
</code></pre></div>
<h3 id="32">3.2 基因组注释<a class="headerlink" href="#32" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>$ <span class="k">for</span> i <span class="k">in</span> <span class="k">$(</span>awk -F<span class="s1">&#39;_&#39;</span> <span class="s1">&#39;!a[$1]++{print}&#39;</span><span class="k">)</span><span class="p">;</span> <span class="k">do</span> prokka --prefix prokka_<span class="nv">$i</span> <span class="nv">$i</span>/contigs.fa<span class="p">;</span> <span class="k">done</span>
</code></pre></div>
<h3 id="33-roary">3.3 用 Roary 进行分析<a class="headerlink" href="#33-roary" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>$ cp prokka_*/<span class="nv">$i</span>.gff gff/ <span class="o">&amp;&amp;</span> <span class="nb">cd</span> gff
$ roary -p -e mafft *.gff
</code></pre></div>
<h3 id="34-harvest">3.4 Harvest<a class="headerlink" href="#34-harvest" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>$ parsnp -g ref.gbk -d data -c -p <span class="m">40</span> -o output
$ harvesttools -i outut/*.ggr -S snp
</code></pre></div>
<h2 id="4-snps">4. SNPs 构建进化树<a class="headerlink" href="#4-snps" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>$ raxml -f a -x <span class="m">12345</span> -p <span class="m">12345</span> -#100 -n ban -T <span class="m">40</span> -s snp -m GTRGAMMA
</code></pre></div>
<h3 id="41-mapping">4.1 基于参考基因组 Mapping 的方法<a class="headerlink" href="#41-mapping" title="Permanent link">&para;</a></h3>
<h4 id="1-snippy">1. Snippy*<a class="headerlink" href="#1-snippy" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="c1"># snippy mapping</span>
$ <span class="k">for</span> i <span class="k">in</span> *.fastq.gz <span class="p">|</span> sort <span class="p">|</span> uniq<span class="p">;</span> <span class="se">\</span>
&gt; <span class="k">do</span> snippy --cpus <span class="m">40</span> --outdir <span class="nv">$i</span> -ref reference.fa <span class="se">\</span>
&gt; --R1 <span class="nv">$i</span>*_1*.fastq.gz --R2 <span class="nv">$i</span>*_2*.fastq.gz<span class="p">;</span> <span class="k">done</span>

<span class="c1"># snippy-core 汇集 snps</span>
$ snippy-core --prefix core-snps SRR* DRR* ERR*

<span class="c1"># raxml 绘制进化树</span>
$ raxml -f a -x <span class="m">12345</span> -p <span class="m">12345</span> -# <span class="m">100</span> -m GTRGAMMA -s core-snps.aln -n ex -T <span class="m">40</span>

<span class="c1"># 将 raxml 进化树结果文件从服务器传到本地电脑，用 figtree 生成图片</span>
$ scp user@server-ip:/path/RAxML_bestTree.ex .
$ figtree RAxML_bestTree.ex <span class="p">&amp;</span>
</code></pre></div>
<h2 id="5">5. 核心基因分析<a class="headerlink" href="#5" title="Permanent link">&para;</a></h2>
<h3 id="51-get_homologues">5.1 get_homologues<a class="headerlink" href="#51-get_homologues" title="Permanent link">&para;</a></h3>
<h2 id="6">6. 相关软件安装<a class="headerlink" href="#6" title="Permanent link">&para;</a></h2>
<p>*****edirect**</p>
<div class="highlight"><pre><span></span><code>$ wget ftp://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/edirect.tar.gz
$ sudo tar zxf edirect.tar.gz -C /opt
$ sudo chown -R root:root /opt/edirect
$ <span class="nb">cd</span> /opt/edirect <span class="o">&amp;&amp;</span> sudo ./setup.sh
$ sudo ln -s /opt/edirect/* /usr/local/sbin/
</code></pre></div>
<p><strong>sra-tools</strong></p>
<div class="highlight"><pre><span></span><code>$ wget http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.7.0/sratoolkit.2.7.0-ubuntu64.tar.gz
$ sudo tar zxf sratoolkit.2.7.0-ubuntu64.tar.gz -C /opt/sratoolkit
$ sudo chown -R root:root /opt/sratoolkit <span class="o">&amp;&amp;</span> <span class="nb">cd</span> /opt/sratoolkit
$ sudo ln -s <span class="sb">`</span><span class="nb">pwd</span><span class="sb">`</span>/bin/* /usr/local/sbin/
</code></pre></div>
<p><strong>bio-awk</strong></p>
<div class="highlight"><pre><span></span><code>$ sudo apt-get install bison
$ git clone https://github.com/lh3/bioawk
$ <span class="nb">cd</span> bioawk <span class="o">&amp;&amp;</span> make
$ sudo cp bioawk /usr/local/sbin
</code></pre></div>
<p><strong>parallel</strong></p>
<div class="highlight"><pre><span></span><code>$ sudo apt install parallel
</code></pre></div>
<p><strong>gplot</strong></p>
<div class="highlight"><pre><span></span><code>$ sudo apt install gnuplot
$ git clone https://github.com/RhysU/gplot
$ sudo cp gplot/gplot /usr/local/sbin
</code></pre></div>
<p><strong>FastQC</strong></p>
<div class="highlight"><pre><span></span><code>$ wget http://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.5.zip
$ sudo unzip fastqc_v0.11.5.zip
$ sudo mv fastqc_v0.11.5 /opt/fastqc <span class="o">&amp;&amp;</span> sudo chown -R root:root /opt/fastqc
$ sudo cp /opt/fastqc/fastqc /usr/local/sbin
</code></pre></div>
<p><strong>Fadapa</strong></p>
<div class="highlight"><pre><span></span><code>$ sudo pip install fadapa
</code></pre></div>
<p><strong>snippy</strong></p>
<div class="highlight"><pre><span></span><code>$ git clone https://github.com/tseemann/snippy
$ sudo cp snippy/bin/* /usr/local/sbin
$ sudo cp snippy/binaries/linux/* /usr/local/sbin
</code></pre></div>
<p><strong>RAxML</strong></p>
<div class="highlight"><pre><span></span><code>$ wget https://github.com/stamatak/standard-RAxML/archive/v8.1.17.tar.gz
$ sudo tar zxf v8.1.17.tar.gz -C /opt/raxml
$ sudo chown -R root:root /opt/raxml <span class="o">&amp;&amp;</span> <span class="nb">cd</span> /opt/raxml
$ sudo make -f Makefile.SSE3.PTHREADS.gcc
$ sudo ln -s <span class="sb">`</span><span class="nb">pwd</span><span class="sb">`</span>/raxmlHPC-PTHREADS-SSE3 /usr/local/raxml
</code></pre></div>
<h2 id="7-reference">7. Reference<a class="headerlink" href="#7-reference" title="Permanent link">&para;</a></h2>
<ol>
<li>edirect help book: <a href="http://www.ncbi.nlm.nih.gov/books/NBK179288/">http://www.ncbi.nlm.nih.gov/books/NBK179288/</a></li>
<li>sra-tools wiki: <a href="https://github.com/ncbi/sra-tools/wiki">https://github.com/ncbi/sra-tools/wiki</a></li>
<li>parallel tutorial: <a href="https://www.biostars.org/p/63816/">https://www.biostars.org/p/63816/</a></li>
<li>gnuplot tutorial: <a href="http://lzz5235.github.io/2016/01/12/gnuplot.html">http://lzz5235.github.io/2016/01/12/gnuplot.html</a></li>
<li>gplot usage: <a href="https://github.com/RhysU/gplot/blob/master/README.md">https://github.com/RhysU/gplot/blob/master/README.md</a></li>
</ol>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid" aria-label="Footer">
        
          <a href="../01_trace-Pae-from-different-source-by-SNP/" class="md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
            </div>
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                01. 细菌SNP溯源分析一
              </div>
            </div>
          </a>
        
        
          <a href="../../C19_Pipelines-and-Workflows/05_workflow-software/" class="md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                05. workflow工具软件
              </div>
            </div>
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2016 - 2021 HZCDC@Lab
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/vendor.18f0862e.min.js"></script>
      <script src="../../../assets/javascripts/bundle.994580cf.min.js"></script><script id="__lang" type="application/json">{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing"}</script>
      
      <script>
        app = initialize({
          base: "../../..",
          features: ['navigation.tabs', 'navigation.sections'],
          search: Object.assign({
            worker: "../../../assets/javascripts/worker/search.9c0e82ba.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script>
      
        <script src="../../../javascripts/config.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>